{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "09453c0f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import gzip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "933d820c",
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import defaultdict\n",
    "mane_gff = '/common/zhangz2lab/shared/genomes/gff/MANE.GRCh38.v1.1.ensembl_genomic.gff.gz'\n",
    "\n",
    "def read_mane(fp):\n",
    "    cds_regions_loaded = defaultdict(list)\n",
    "    with gzip.open(fp, 'rt') as f:  # Open the GFF3 file\n",
    "        for line in f:\n",
    "            if not line.startswith('#'):  # Skip header lines\n",
    "                parts = line.strip().split('\\t')\n",
    "                if not parts[0] in ['chr%i'%i for i in range(1,23)] + ['chrX', 'chrY']:\n",
    "                    continue\n",
    "                if parts[2] == 'CDS':  # Focus on CDS entries\n",
    "                    attributes = parts[8]\n",
    "                    gene_id = [attr for attr in attributes.split(';') if 'gene_name' in attr][0].split('=')[1]\n",
    "                    strand = parts[6]\n",
    "                    start = parts[3]\n",
    "                    end = parts[4]\n",
    "                    reading_frame = parts[7]\n",
    "                    chrom = parts[0]\n",
    "                    cds_regions_loaded[gene_id].append((int(start), int(end), int(reading_frame), strand, chrom))\n",
    "                    \n",
    "    return cds_regions_loaded\n",
    "\n",
    "cds_regions_loaded = read_mane(mane_gff)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "44b7219b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[(3069260, 3069296, 0, '+', 'chr1'), (3186125, 3186474, 2, '+', 'chr1'), (3244087, 3244137, 0, '+', 'chr1'), (3385152, 3385286, 0, '+', 'chr1'), (3396491, 3396593, 0, '+', 'chr1'), (3402791, 3402998, 2, '+', 'chr1'), (3404739, 3404886, 1, '+', 'chr1'), (3405495, 3405648, 0, '+', 'chr1'), (3411384, 3412800, 2, '+', 'chr1'), (3414560, 3414647, 1, '+', 'chr1'), (3417828, 3417997, 0, '+', 'chr1'), (3418667, 3418744, 1, '+', 'chr1'), (3425581, 3425750, 1, '+', 'chr1'), (3426051, 3426225, 2, '+', 'chr1'), (3430872, 3431108, 1, '+', 'chr1'), (3431966, 3432140, 1, '+', 'chr1'), (3433677, 3433811, 0, '+', 'chr1')]\n"
     ]
    }
   ],
   "source": [
    "print(cds_regions_loaded.get(\"PRDM16\"))  # Example usage\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b23dddca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>cln_disease</th>\n",
       "      <th>cln_disease_id</th>\n",
       "      <th>cln_significance</th>\n",
       "      <th>cln_gene</th>\n",
       "      <th>cln_review</th>\n",
       "      <th>cln_hgvs</th>\n",
       "      <th>cln_disease_dbs</th>\n",
       "      <th>cln_clnvi</th>\n",
       "      <th>cln_af_exac</th>\n",
       "      <th>cln_af_esp</th>\n",
       "      <th>cln_af_tgp</th>\n",
       "      <th>cln_molecular_consequence</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>925952</td>\n",
       "      <td>1019397</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>MedGen:CN517202</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>NC_000001.11:g.925952G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>925956</td>\n",
       "      <td>1543320</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>MedGen:CN517202</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>NC_000001.11:g.925956C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>925969</td>\n",
       "      <td>1648427</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>MedGen:CN517202</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>NC_000001.11:g.925969C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>925976</td>\n",
       "      <td>1362713</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>MedGen:CN517202</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>NC_000001.11:g.925976T&gt;C</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>925986</td>\n",
       "      <td>1568423</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>MedGen:CN517202</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>NC_000001.11:g.925986C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS       ID REF ALT QUAL FILTER   cln_disease   cln_disease_id  \\\n",
       "0     1  925952  1019397   G   A    .      .  not_provided  MedGen:CN517202   \n",
       "1     1  925956  1543320   C   T    .      .  not_provided  MedGen:CN517202   \n",
       "2     1  925969  1648427   C   T    .      .  not_provided  MedGen:CN517202   \n",
       "3     1  925976  1362713   T   C    .      .  not_provided  MedGen:CN517202   \n",
       "4     1  925986  1568423   C   T    .      .  not_provided  MedGen:CN517202   \n",
       "\n",
       "         cln_significance       cln_gene                           cln_review  \\\n",
       "0  Uncertain_significance  SAMD11:148398  criteria_provided,_single_submitter   \n",
       "1           Likely_benign  SAMD11:148398  criteria_provided,_single_submitter   \n",
       "2           Likely_benign  SAMD11:148398  criteria_provided,_single_submitter   \n",
       "3  Uncertain_significance  SAMD11:148398  criteria_provided,_single_submitter   \n",
       "4           Likely_benign  SAMD11:148398  criteria_provided,_single_submitter   \n",
       "\n",
       "                   cln_hgvs cln_disease_dbs cln_clnvi cln_af_exac cln_af_esp  \\\n",
       "0  NC_000001.11:g.925952G>A            None      None        None       None   \n",
       "1  NC_000001.11:g.925956C>T            None      None        None       None   \n",
       "2  NC_000001.11:g.925969C>T            None      None        None       None   \n",
       "3  NC_000001.11:g.925976T>C            None      None        None       None   \n",
       "4  NC_000001.11:g.925986C>T            None      None        None       None   \n",
       "\n",
       "  cln_af_tgp      cln_molecular_consequence gene_name  \n",
       "0       None    SO:0001583|missense_variant    SAMD11  \n",
       "1       None  SO:0001819|synonymous_variant    SAMD11  \n",
       "2       None    SO:0001583|missense_variant    SAMD11  \n",
       "3       None    SO:0001583|missense_variant    SAMD11  \n",
       "4       None  SO:0001819|synonymous_variant    SAMD11  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def parse_string_to_dict(info_string):\n",
    "    \"\"\"Convert a VCF INFO field string to a dictionary.\"\"\"\n",
    "    info_dict = {}\n",
    "    for item in info_string.split(';'):\n",
    "        if '=' in item:\n",
    "            key, value = item.split('=', 1)\n",
    "            info_dict[key] = value\n",
    "    return info_dict\n",
    "\n",
    "def parse_clinvar_to_dataframe(filepath):\n",
    "    with gzip.open(filepath, 'rt') as file:\n",
    "        data = []  # List to store combined information for each variant\n",
    "        for line in file:\n",
    "            if line.startswith('#'):\n",
    "                continue  # Skip header lines\n",
    "            fields = line.strip().split('\\t')\n",
    "            info_field = fields[7]  # INFO field is the 8th field in VCF format\n",
    "            info_dict = parse_string_to_dict(info_field)\n",
    "            \n",
    "            # Combine first 7 fields with extracted INFO data\n",
    "            variant_info = fields[:7] + [\n",
    "                info_dict.get('CLNDN'),\n",
    "                info_dict.get('CLNDISDB'),\n",
    "                info_dict.get('CLNSIG'),\n",
    "                info_dict.get('GENEINFO'),\n",
    "                info_dict.get('CLNREVSTAT'),\n",
    "                info_dict.get('CLNHGVS'),\n",
    "                info_dict.get('CLNDISDBINCL'),\n",
    "                info_dict.get('CLNVI'),\n",
    "                info_dict.get('AF_EXAC'),\n",
    "                info_dict.get('AF_ESP'),\n",
    "                info_dict.get('AF_TGP'),\n",
    "                info_dict.get('MC'),\n",
    "            ]\n",
    "            \n",
    "            data.append(variant_info)\n",
    "        \n",
    "        # Convert the accumulated data into a pandas DataFrame\n",
    "        columns = ['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER',\n",
    "                   'cln_disease', 'cln_disease_id', 'cln_significance', 'cln_gene',\n",
    "                   'cln_review', 'cln_hgvs', 'cln_disease_dbs', 'cln_clnvi',\n",
    "                   'cln_af_exac', 'cln_af_esp', 'cln_af_tgp', 'cln_molecular_consequence']\n",
    "        df = pd.DataFrame(data, columns=columns)\n",
    "        \n",
    "        return df\n",
    "\n",
    "# Example usage\n",
    "filepath = '/common/zhangz2lab/fzzhang/knowledge_graphs/wen-chyu/data/variant/clinvar/clinvar.vcf.gz' \n",
    "df = parse_clinvar_to_dataframe(filepath)\n",
    "\n",
    "df = df.dropna(subset=['cln_gene'])\n",
    "df['gene_name'] = df.apply(lambda row: row['cln_gene'].split(':')[0], axis=1)\n",
    "\n",
    "# Display the first few rows of the DataFrame\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "381c310b",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>cln_disease</th>\n",
       "      <th>cln_disease_id</th>\n",
       "      <th>cln_significance</th>\n",
       "      <th>cln_gene</th>\n",
       "      <th>cln_review</th>\n",
       "      <th>cln_hgvs</th>\n",
       "      <th>cln_disease_dbs</th>\n",
       "      <th>cln_clnvi</th>\n",
       "      <th>cln_af_exac</th>\n",
       "      <th>cln_af_esp</th>\n",
       "      <th>cln_af_tgp</th>\n",
       "      <th>cln_molecular_consequence</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>65750</th>\n",
       "      <td>1</td>\n",
       "      <td>115726998</td>\n",
       "      <td>35772</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Cardiac_arrhythmia|Cardiomyopathy|Catecholamin...</td>\n",
       "      <td>EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>CASQ2:845</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>NC_000001.11:g.115726998T&gt;C</td>\n",
       "      <td>MedGen:CN169374</td>\n",
       "      <td>HGMD:CM106915|ARUP_Laboratories,_Molecular_Gen...</td>\n",
       "      <td>0.00829</td>\n",
       "      <td>0.02660</td>\n",
       "      <td>0.02835</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>CASQ2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85241</th>\n",
       "      <td>1</td>\n",
       "      <td>169130013</td>\n",
       "      <td>35698</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Cardiac_arrhythmia</td>\n",
       "      <td>EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>ATP1B1:481|NME7:29922</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>NC_000001.11:g.169130013C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>ATP1B1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122394</th>\n",
       "      <td>1</td>\n",
       "      <td>237270595</td>\n",
       "      <td>178876</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Cardiac_arrhythmia|Cardiomyopathy|Catecholamin...</td>\n",
       "      <td>EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>RYR2:6262</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>NC_000001.11:g.237270595G&gt;C</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>RYR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123625</th>\n",
       "      <td>1</td>\n",
       "      <td>237550628</td>\n",
       "      <td>201235</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Cardiomyopathy|Arrhythmogenic_right_ventricula...</td>\n",
       "      <td>Human_Phenotype_Ontology:HP:0001638,MONDO:MOND...</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>RYR2:6262</td>\n",
       "      <td>criteria_provided,_conflicting_interpretations</td>\n",
       "      <td>NC_000001.11:g.237550628C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.00017</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>RYR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123665</th>\n",
       "      <td>1</td>\n",
       "      <td>237566603</td>\n",
       "      <td>36740</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Primary_dilated_cardiomyopathy|Cardiac_arrhyth...</td>\n",
       "      <td>EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>RYR2:6262</td>\n",
       "      <td>criteria_provided,_conflicting_interpretations</td>\n",
       "      <td>NC_000001.11:g.237566603G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.00015</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>RYR2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS      ID REF ALT QUAL FILTER  \\\n",
       "65750      1  115726998   35772   T   C    .      .   \n",
       "85241      1  169130013   35698   C   T    .      .   \n",
       "122394     1  237270595  178876   G   C    .      .   \n",
       "123625     1  237550628  201235   C   T    .      .   \n",
       "123665     1  237566603   36740   G   A    .      .   \n",
       "\n",
       "                                              cln_disease  \\\n",
       "65750   Cardiac_arrhythmia|Cardiomyopathy|Catecholamin...   \n",
       "85241                                  Cardiac_arrhythmia   \n",
       "122394  Cardiac_arrhythmia|Cardiomyopathy|Catecholamin...   \n",
       "123625  Cardiomyopathy|Arrhythmogenic_right_ventricula...   \n",
       "123665  Primary_dilated_cardiomyopathy|Cardiac_arrhyth...   \n",
       "\n",
       "                                           cln_disease_id  \\\n",
       "65750   EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
       "85241   EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
       "122394  EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
       "123625  Human_Phenotype_Ontology:HP:0001638,MONDO:MOND...   \n",
       "123665  EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...   \n",
       "\n",
       "                                    cln_significance               cln_gene  \\\n",
       "65750                           Benign/Likely_benign              CASQ2:845   \n",
       "85241                         Uncertain_significance  ATP1B1:481|NME7:29922   \n",
       "122394                        Uncertain_significance              RYR2:6262   \n",
       "123625  Conflicting_interpretations_of_pathogenicity              RYR2:6262   \n",
       "123665  Conflicting_interpretations_of_pathogenicity              RYR2:6262   \n",
       "\n",
       "                                               cln_review  \\\n",
       "65750   criteria_provided,_multiple_submitters,_no_con...   \n",
       "85241                      no_assertion_criteria_provided   \n",
       "122394  criteria_provided,_multiple_submitters,_no_con...   \n",
       "123625     criteria_provided,_conflicting_interpretations   \n",
       "123665     criteria_provided,_conflicting_interpretations   \n",
       "\n",
       "                           cln_hgvs  cln_disease_dbs  \\\n",
       "65750   NC_000001.11:g.115726998T>C  MedGen:CN169374   \n",
       "85241   NC_000001.11:g.169130013C>T             None   \n",
       "122394  NC_000001.11:g.237270595G>C             None   \n",
       "123625  NC_000001.11:g.237550628C>T             None   \n",
       "123665  NC_000001.11:g.237566603G>A             None   \n",
       "\n",
       "                                                cln_clnvi cln_af_exac  \\\n",
       "65750   HGMD:CM106915|ARUP_Laboratories,_Molecular_Gen...     0.00829   \n",
       "85241                                                None        None   \n",
       "122394                                               None        None   \n",
       "123625                                               None     0.00017   \n",
       "123665                                               None     0.00015   \n",
       "\n",
       "       cln_af_esp cln_af_tgp    cln_molecular_consequence gene_name  \n",
       "65750     0.02660    0.02835  SO:0001583|missense_variant     CASQ2  \n",
       "85241        None       None  SO:0001583|missense_variant    ATP1B1  \n",
       "122394       None       None  SO:0001583|missense_variant      RYR2  \n",
       "123625       None       None  SO:0001583|missense_variant      RYR2  \n",
       "123665       None       None  SO:0001583|missense_variant      RYR2  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_df = df[\n",
    "    (df['cln_disease'].str.contains('arrhythmia', case=False, na=False)) & \n",
    "     (df['cln_molecular_consequence'].str.contains('missense', case=False, na=False))\n",
    "]\n",
    "\n",
    "\n",
    "# Display the filtered DataFrame\n",
    "filtered_df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0640c788",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    chrom      start        end         original_position  score\n",
      "0      11    2445116    2445117     chr11:2466347-2466347      1\n",
      "1      11    2445418    2445419     chr11:2466649-2466649      1\n",
      "2      11    2445446    2445447     chr11:2466677-2466677      1\n",
      "3      11    2445447    2445448     chr11:2466678-2466678      1\n",
      "4      11    2445470    2445471     chr11:2466701-2466701      1\n",
      "..    ...        ...        ...                       ...    ...\n",
      "303     3   38575394   38575395    chr3:38616886-38616886      1\n",
      "304     7  150950274  150950275  chr7:150647363-150647363      1\n",
      "305    12    2585890    2585891     chr12:2695057-2695057      1\n",
      "306    12    2567571    2567572     chr12:2676738-2676738      1\n",
      "307     3   38613771   38613772    chr3:38655263-38655263      1\n",
      "\n",
      "[308 rows x 5 columns]\n",
      "    CHROM       POS       ID REF ALT QUAL FILTER  \\\n",
      "0       3  38550338    68027   G   A    .      .   \n",
      "1       3  38550409    68020   A   C    .      .   \n",
      "2       3  38550499    68015   C   T    .      .   \n",
      "3       3  38550577   582721   G   A    .      .   \n",
      "4       3  38550602     9381   C   T    .      .   \n",
      "..    ...       ...      ...  ..  ..  ...    ...   \n",
      "114    11   2777000    53007   T   C    .      .   \n",
      "115    11   2847771    67057   C   T    .      .   \n",
      "116    11   2847803    67059   G   A    .      .   \n",
      "117    11   2847992  1172315   G   A    .      .   \n",
      "118    12   2691054   216482   A   G    .      .   \n",
      "\n",
      "                                           cln_disease  \\\n",
      "0    Cardiac_arrhythmia|Long_QT_syndrome_3|Brugada_...   \n",
      "1    Cardiac_arrhythmia|Progressive_familial_heart_...   \n",
      "2    Cardiac_arrhythmia|Long_QT_syndrome|Brugada_sy...   \n",
      "3    Primary_dilated_cardiomyopathy|Cardiac_arrhyth...   \n",
      "4    Cardiac_arrhythmia|Long_QT_syndrome|Brugada_sy...   \n",
      "..                                                 ...   \n",
      "114  Cardiac_arrhythmia|Long_QT_syndrome|Congenital...   \n",
      "115  Cardiac_arrhythmia|Long_QT_syndrome|Congenital...   \n",
      "116  Cardiac_arrhythmia|Long_QT_syndrome|Congenital...   \n",
      "117                                 Cardiac_arrhythmia   \n",
      "118  Cardiac_arrhythmia|Long_QT_syndrome|Timothy_sy...   \n",
      "\n",
      "                                        cln_disease_id  \\\n",
      "0    EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
      "1    EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
      "2    EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
      "3    EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...   \n",
      "4    EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
      "..                                                 ...   \n",
      "114  EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
      "115  EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
      "116  EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
      "117  EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
      "118  EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
      "\n",
      "                                 cln_significance  ... cln_af_exac cln_af_esp  \\\n",
      "0                          Uncertain_significance  ...     0.00002       None   \n",
      "1    Conflicting_interpretations_of_pathogenicity  ...     0.00031       None   \n",
      "2    Conflicting_interpretations_of_pathogenicity  ...     0.00006    0.00016   \n",
      "3    Conflicting_interpretations_of_pathogenicity  ...     0.00007    0.00008   \n",
      "4    Conflicting_interpretations_of_pathogenicity  ...     0.00004       None   \n",
      "..                                            ...  ...         ...        ...   \n",
      "114                  Pathogenic/Likely_pathogenic  ...        None       None   \n",
      "115  Conflicting_interpretations_of_pathogenicity  ...     0.00034    0.00054   \n",
      "116  Conflicting_interpretations_of_pathogenicity  ...     0.00022    0.00031   \n",
      "117                        Uncertain_significance  ...        None       None   \n",
      "118  Conflicting_interpretations_of_pathogenicity  ...     0.00079    0.00058   \n",
      "\n",
      "    cln_af_tgp    cln_molecular_consequence gene_name chrom     start  \\\n",
      "0         None  SO:0001583|missense_variant     SCN5A     3  38550337   \n",
      "1      0.00040  SO:0001583|missense_variant     SCN5A     3  38550408   \n",
      "2      0.00040  SO:0001583|missense_variant     SCN5A     3  38550498   \n",
      "3         None  SO:0001583|missense_variant     SCN5A     3  38550576   \n",
      "4         None  SO:0001583|missense_variant     SCN5A     3  38550601   \n",
      "..         ...                          ...       ...   ...       ...   \n",
      "114       None  SO:0001583|missense_variant     KCNQ1    11   2776999   \n",
      "115    0.00020  SO:0001583|missense_variant     KCNQ1    11   2847770   \n",
      "116    0.00020  SO:0001583|missense_variant     KCNQ1    11   2847802   \n",
      "117       None  SO:0001583|missense_variant     KCNQ1    11   2847991   \n",
      "118    0.00020  SO:0001583|missense_variant   CACNA1C    12   2691053   \n",
      "\n",
      "          end       original_position score  \n",
      "0    38550338  chr3:38591829-38591829     1  \n",
      "1    38550409  chr3:38591900-38591900     1  \n",
      "2    38550499  chr3:38591990-38591990     1  \n",
      "3    38550577  chr3:38592068-38592068     1  \n",
      "4    38550602  chr3:38592093-38592093     1  \n",
      "..        ...                     ...   ...  \n",
      "114   2777000   chr11:2798230-2798230     1  \n",
      "115   2847771   chr11:2869001-2869001     1  \n",
      "116   2847803   chr11:2869033-2869033     1  \n",
      "117   2847992   chr11:2869222-2869222     1  \n",
      "118   2691054   chr12:2800220-2800220     1  \n",
      "\n",
      "[119 rows x 25 columns]\n",
      "0      G\n",
      "1      A\n",
      "2      C\n",
      "3      G\n",
      "4      C\n",
      "      ..\n",
      "114    T\n",
      "115    C\n",
      "116    G\n",
      "117    G\n",
      "118    A\n",
      "Name: REF, Length: 119, dtype: object\n",
      "  CHROM        POS      ID REF ALT QUAL FILTER  \\\n",
      "0     1  115726998   35772   T   C    .      .   \n",
      "1     1  169130013   35698   C   T    .      .   \n",
      "2     1  237270595  178876   G   C    .      .   \n",
      "3     1  237550628  201235   C   T    .      .   \n",
      "4     1  237566603   36740   G   A    .      .   \n",
      "\n",
      "                                         cln_disease  \\\n",
      "0  Cardiac_arrhythmia|Cardiomyopathy|Catecholamin...   \n",
      "1                                 Cardiac_arrhythmia   \n",
      "2  Cardiac_arrhythmia|Cardiomyopathy|Catecholamin...   \n",
      "3  Cardiomyopathy|Arrhythmogenic_right_ventricula...   \n",
      "4  Primary_dilated_cardiomyopathy|Cardiac_arrhyth...   \n",
      "\n",
      "                                      cln_disease_id  \\\n",
      "0  EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
      "1  EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
      "2  EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
      "3  Human_Phenotype_Ontology:HP:0001638,MONDO:MOND...   \n",
      "4  EFO:EFO_0000407,Human_Phenotype_Ontology:HP:00...   \n",
      "\n",
      "                               cln_significance  ... cln_af_exac cln_af_esp  \\\n",
      "0                          Benign/Likely_benign  ...     0.00829    0.02660   \n",
      "1                        Uncertain_significance  ...        None       None   \n",
      "2                        Uncertain_significance  ...        None       None   \n",
      "3  Conflicting_interpretations_of_pathogenicity  ...     0.00017       None   \n",
      "4  Conflicting_interpretations_of_pathogenicity  ...     0.00015       None   \n",
      "\n",
      "  cln_af_tgp    cln_molecular_consequence gene_name chrom start end  \\\n",
      "0    0.02835  SO:0001583|missense_variant     CASQ2   NaN   NaN NaN   \n",
      "1       None  SO:0001583|missense_variant    ATP1B1   NaN   NaN NaN   \n",
      "2       None  SO:0001583|missense_variant      RYR2   NaN   NaN NaN   \n",
      "3       None  SO:0001583|missense_variant      RYR2   NaN   NaN NaN   \n",
      "4       None  SO:0001583|missense_variant      RYR2   NaN   NaN NaN   \n",
      "\n",
      "  original_position score  \n",
      "0               NaN   NaN  \n",
      "1               NaN   NaN  \n",
      "2               NaN   NaN  \n",
      "3               NaN   NaN  \n",
      "4               NaN   NaN  \n",
      "\n",
      "[5 rows x 25 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3730043/1895868383.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  filtered_df['CHROM'] = filtered_df['CHROM'].astype(str)\n",
      "/tmp/ipykernel_3730043/1895868383.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  filtered_df['POS'] = filtered_df['POS'].astype(int)\n"
     ]
    }
   ],
   "source": [
    "new_bed_df = pd.read_csv('output_bed_arm.csv', index_col=0)\n",
    "# Remove 'chr' prefix from 'chrom' column in new_bed_df\n",
    "new_bed_df['chrom'] = new_bed_df['chrom'].str.replace('chr', '')\n",
    "print(new_bed_df)\n",
    "# Convert 'chrom' columns to strings in both DataFrames if they are not already\n",
    "filtered_df['CHROM'] = filtered_df['CHROM'].astype(str)\n",
    "new_bed_df['chrom'] = new_bed_df['chrom'].astype(str)\n",
    "\n",
    "# Ensure 'POS' in filtered_df and 'end' in new_bed_df are integers\n",
    "filtered_df['POS'] = filtered_df['POS'].astype(int)\n",
    "new_bed_df['end'] = new_bed_df['end'].astype(int)\n",
    "\n",
    "# Attempt the merge again\n",
    "matches = filtered_df.merge(new_bed_df, left_on=['CHROM', 'POS'], right_on=['chrom', 'end'])\n",
    "print(matches)\n",
    "print(matches['REF'])\n",
    "\n",
    "# Perform the merge to identify rows to keep (\"left_only\") and use suffixes to handle overlapping column names\n",
    "# We'll use suffixes that are unlikely to cause conflicts and are easy to filter out\n",
    "merged_df = pd.merge(filtered_df, matches, on=['CHROM', 'POS'], how='left', indicator=True, suffixes=('', '_drop'))\n",
    "\n",
    "# Filter to keep only rows from filtered_df that don't have matches in new_bed_df\n",
    "filtered_df_no_matches = merged_df[merged_df['_merge'] == 'left_only']\n",
    "\n",
    "# Now, remove columns that were added from 'matches' (those with '_drop' suffix) and the '_merge' column\n",
    "# This is done by filtering out columns that end with '_drop' and dropping the '_merge' column\n",
    "filtered_df_no_matches = filtered_df_no_matches.loc[:, ~filtered_df_no_matches.columns.str.endswith('_drop')].drop(columns=['_merge'])\n",
    "\n",
    "\n",
    "filtered_df = filtered_df_no_matches \n",
    "# Now, filtered_df_no_matches should have the original structure without \"_x\" or \"_y\" suffixes\n",
    "print(filtered_df_no_matches.head())\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "b9615ef5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>cln_disease</th>\n",
       "      <th>cln_disease_id</th>\n",
       "      <th>cln_significance</th>\n",
       "      <th>...</th>\n",
       "      <th>cln_af_exac</th>\n",
       "      <th>cln_af_esp</th>\n",
       "      <th>cln_af_tgp</th>\n",
       "      <th>cln_molecular_consequence</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>original_position</th>\n",
       "      <th>score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>169130013</td>\n",
       "      <td>35698</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Cardiac_arrhythmia</td>\n",
       "      <td>EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>ATP1B1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>237270595</td>\n",
       "      <td>178876</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Cardiac_arrhythmia|Cardiomyopathy|Catecholamin...</td>\n",
       "      <td>EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>RYR2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>237631470</td>\n",
       "      <td>1691305</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Arrhythmogenic_right_ventricular_dysplasia_2|C...</td>\n",
       "      <td>MONDO:MONDO:0010975,MedGen:C1832931,OMIM:60099...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>RYR2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>237784793</td>\n",
       "      <td>201343</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Catecholaminergic_polymorphic_ventricular_tach...</td>\n",
       "      <td>MONDO:MONDO:0017990,MedGen:C5574922,OMIM:PS604...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>RYR2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1</td>\n",
       "      <td>237795312</td>\n",
       "      <td>852605</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Catecholaminergic_polymorphic_ventricular_tach...</td>\n",
       "      <td>MONDO:MONDO:0017990,MedGen:C5574922,OMIM:PS604...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>RYR2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1345</th>\n",
       "      <td>22</td>\n",
       "      <td>20052582</td>\n",
       "      <td>1339009</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Recurrent_metabolic_encephalomyopathic_crises-...</td>\n",
       "      <td>MONDO:MONDO:0018820,MedGen:C5567524,OMIM:61687...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>TANGO2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1346</th>\n",
       "      <td>22</td>\n",
       "      <td>20052584</td>\n",
       "      <td>1312498</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Recurrent_metabolic_encephalomyopathic_crises-...</td>\n",
       "      <td>MONDO:MONDO:0018820,MedGen:C5567524,OMIM:61687...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>TANGO2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1347</th>\n",
       "      <td>22</td>\n",
       "      <td>20053530</td>\n",
       "      <td>1709923</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Recurrent_metabolic_encephalomyopathic_crises-...</td>\n",
       "      <td>MONDO:MONDO:0018820,MedGen:C5567524,OMIM:61687...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>TANGO2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1348</th>\n",
       "      <td>22</td>\n",
       "      <td>20055992</td>\n",
       "      <td>1696470</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Recurrent_metabolic_encephalomyopathic_crises-...</td>\n",
       "      <td>MONDO:MONDO:0018820,MedGen:C5567524,OMIM:61687...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>TANGO2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1350</th>\n",
       "      <td>22</td>\n",
       "      <td>20063435</td>\n",
       "      <td>1301615</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Recurrent_metabolic_encephalomyopathic_crises-...</td>\n",
       "      <td>MONDO:MONDO:0018820,MedGen:C5567524,OMIM:61687...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "      <td>TANGO2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>888 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS       ID REF ALT QUAL FILTER  \\\n",
       "1        1  169130013    35698   C   T    .      .   \n",
       "2        1  237270595   178876   G   C    .      .   \n",
       "6        1  237631470  1691305   A   C    .      .   \n",
       "9        1  237784793   201343   G   A    .      .   \n",
       "10       1  237795312   852605   A   C    .      .   \n",
       "...    ...        ...      ...  ..  ..  ...    ...   \n",
       "1345    22   20052582  1339009   G   A    .      .   \n",
       "1346    22   20052584  1312498   G   T    .      .   \n",
       "1347    22   20053530  1709923   A   G    .      .   \n",
       "1348    22   20055992  1696470   G   A    .      .   \n",
       "1350    22   20063435  1301615   G   A    .      .   \n",
       "\n",
       "                                            cln_disease  \\\n",
       "1                                    Cardiac_arrhythmia   \n",
       "2     Cardiac_arrhythmia|Cardiomyopathy|Catecholamin...   \n",
       "6     Arrhythmogenic_right_ventricular_dysplasia_2|C...   \n",
       "9     Catecholaminergic_polymorphic_ventricular_tach...   \n",
       "10    Catecholaminergic_polymorphic_ventricular_tach...   \n",
       "...                                                 ...   \n",
       "1345  Recurrent_metabolic_encephalomyopathic_crises-...   \n",
       "1346  Recurrent_metabolic_encephalomyopathic_crises-...   \n",
       "1347  Recurrent_metabolic_encephalomyopathic_crises-...   \n",
       "1348  Recurrent_metabolic_encephalomyopathic_crises-...   \n",
       "1350  Recurrent_metabolic_encephalomyopathic_crises-...   \n",
       "\n",
       "                                         cln_disease_id  \\\n",
       "1     EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
       "2     EFO:EFO_0004269,Human_Phenotype_Ontology:HP:00...   \n",
       "6     MONDO:MONDO:0010975,MedGen:C1832931,OMIM:60099...   \n",
       "9     MONDO:MONDO:0017990,MedGen:C5574922,OMIM:PS604...   \n",
       "10    MONDO:MONDO:0017990,MedGen:C5574922,OMIM:PS604...   \n",
       "...                                                 ...   \n",
       "1345  MONDO:MONDO:0018820,MedGen:C5567524,OMIM:61687...   \n",
       "1346  MONDO:MONDO:0018820,MedGen:C5567524,OMIM:61687...   \n",
       "1347  MONDO:MONDO:0018820,MedGen:C5567524,OMIM:61687...   \n",
       "1348  MONDO:MONDO:0018820,MedGen:C5567524,OMIM:61687...   \n",
       "1350  MONDO:MONDO:0018820,MedGen:C5567524,OMIM:61687...   \n",
       "\n",
       "            cln_significance  ... cln_af_exac cln_af_esp cln_af_tgp  \\\n",
       "1     Uncertain_significance  ...        None       None       None   \n",
       "2     Uncertain_significance  ...        None       None       None   \n",
       "6     Uncertain_significance  ...        None       None       None   \n",
       "9     Uncertain_significance  ...        None       None       None   \n",
       "10    Uncertain_significance  ...        None       None       None   \n",
       "...                      ...  ...         ...        ...        ...   \n",
       "1345  Uncertain_significance  ...        None       None       None   \n",
       "1346  Uncertain_significance  ...        None       None       None   \n",
       "1347  Uncertain_significance  ...        None       None       None   \n",
       "1348  Uncertain_significance  ...        None       None       None   \n",
       "1350  Uncertain_significance  ...        None       None       None   \n",
       "\n",
       "                              cln_molecular_consequence gene_name chrom start  \\\n",
       "1                           SO:0001583|missense_variant    ATP1B1   NaN   NaN   \n",
       "2                           SO:0001583|missense_variant      RYR2   NaN   NaN   \n",
       "6                           SO:0001583|missense_variant      RYR2   NaN   NaN   \n",
       "9                           SO:0001583|missense_variant      RYR2   NaN   NaN   \n",
       "10                          SO:0001583|missense_variant      RYR2   NaN   NaN   \n",
       "...                                                 ...       ...   ...   ...   \n",
       "1345  SO:0001583|missense_variant,SO:0001619|non-cod...    TANGO2   NaN   NaN   \n",
       "1346  SO:0001583|missense_variant,SO:0001619|non-cod...    TANGO2   NaN   NaN   \n",
       "1347  SO:0001583|missense_variant,SO:0001619|non-cod...    TANGO2   NaN   NaN   \n",
       "1348  SO:0001583|missense_variant,SO:0001619|non-cod...    TANGO2   NaN   NaN   \n",
       "1350  SO:0001583|missense_variant,SO:0001619|non-cod...    TANGO2   NaN   NaN   \n",
       "\n",
       "     end original_position score  \n",
       "1    NaN               NaN   NaN  \n",
       "2    NaN               NaN   NaN  \n",
       "6    NaN               NaN   NaN  \n",
       "9    NaN               NaN   NaN  \n",
       "10   NaN               NaN   NaN  \n",
       "...   ..               ...   ...  \n",
       "1345 NaN               NaN   NaN  \n",
       "1346 NaN               NaN   NaN  \n",
       "1347 NaN               NaN   NaN  \n",
       "1348 NaN               NaN   NaN  \n",
       "1350 NaN               NaN   NaN  \n",
       "\n",
       "[888 rows x 25 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter the DataFrame to include only rows with 'cln_significance' containing \"benign\" or \"pathogenic\"\n",
    "filtered_df = filtered_df[filtered_df['cln_significance'].str.contains('uncertain', case=False)]\n",
    "\n",
    "# Filter out conflicting annotations\n",
    "filtered_df = filtered_df[~filtered_df['cln_significance'].str.contains('conflicting', case=False)]\n",
    "\n",
    "\n",
    "# Count rows where 'cln_significance' contains \"benign\"\n",
    "#benign_count = filtered_df[filtered_df['cln_significance'].str.contains('benign', case=False)].shape[0]\n",
    "\n",
    "# Assuming you want to count rows where 'cln_significance' contains \"pathogenic\" (regardless of 'cln_molecular_consequence')\n",
    "#pathogenic_count = filtered_df[filtered_df['cln_significance'].str.contains('pathogenic', case=False)].shape[0]\n",
    "\n",
    "#print(f\"Number of 'benign' rows: {benign_count}\")\n",
    "#print(f\"Number of 'pathogenic' rows: {pathogenic_count}\")\n",
    "filtered_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c526ff84",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unique CHROM values in the DataFrame:\n",
      "1\n",
      "3\n",
      "4\n",
      "7\n",
      "9\n",
      "11\n",
      "12\n",
      "14\n",
      "15\n",
      "17\n",
      "22\n",
      "Total number of unique CHROM values: 11\n"
     ]
    }
   ],
   "source": [
    "# Get unique CHROM values from the DataFrame\n",
    "unique_chrom_values = filtered_df['CHROM'].unique()\n",
    "\n",
    "# Print all unique CHROM values\n",
    "print(\"Unique CHROM values in the DataFrame:\")\n",
    "for chrom in unique_chrom_values:\n",
    "    print(chrom)\n",
    "\n",
    "# Count the number of unique CHROM values\n",
    "number_of_unique_chrom_values = len(unique_chrom_values)\n",
    "print(f\"Total number of unique CHROM values: {number_of_unique_chrom_values}\")\n",
    "filtered_df = filtered_df[~filtered_df['CHROM'].isin(['MT', 'NW_009646201.1'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4466316a",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Seq import Seq\n",
    "\n",
    "# Function to translate DNA sequence to protein sequence\n",
    "def translate_dna_to_protein(dna_sequence):\n",
    "    dna_seq = Seq(dna_sequence)\n",
    "    protein_seq = dna_seq.translate(to_stop=True)  # to_stop=True will stop at the first stop codon\n",
    "    return str(protein_seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "77981112",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get proteins for each gene\n",
    "from pyfaidx import Fasta\n",
    "from Bio.Seq import Seq\n",
    "from tqdm import tqdm\n",
    "\n",
    "reference_genome = Fasta('/common/zhangz2lab/zhanh/Clinvar/hg38.fa')\n",
    "\n",
    "def reverse_complement(seq):\n",
    "    complement = {'A': 'T', 'T': 'A', 'G': 'C', 'C': 'G'}\n",
    "    return ''.join(complement.get(base, base) for base in reversed(seq))\n",
    "\n",
    "\n",
    "codon_table = {\n",
    "    'ATA':'I', 'ATC':'I', 'ATT':'I', 'ATG':'M',\n",
    "    'ACA':'T', 'ACC':'T', 'ACG':'T', 'ACT':'T',\n",
    "    'AAC':'N', 'AAT':'N', 'AAA':'K', 'AAG':'K',\n",
    "    'AGC':'S', 'AGT':'S', 'AGA':'R', 'AGG':'R',\n",
    "    'CTA':'L', 'CTC':'L', 'CTG':'L', 'CTT':'L',\n",
    "    'CCA':'P', 'CCC':'P', 'CCG':'P', 'CCT':'P',\n",
    "    'CAC':'H', 'CAT':'H', 'CAA':'Q', 'CAG':'Q',\n",
    "    'CGA':'R', 'CGC':'R', 'CGG':'R', 'CGT':'R',\n",
    "    'GTA':'V', 'GTC':'V', 'GTG':'V', 'GTT':'V',\n",
    "    'GCA':'A', 'GCC':'A', 'GCG':'A', 'GCT':'A',\n",
    "    'GAC':'D', 'GAT':'D', 'GAA':'E', 'GAG':'E',\n",
    "    'GGA':'G', 'GGC':'G', 'GGG':'G', 'GGT':'G',\n",
    "    'TCA':'S', 'TCC':'S', 'TCG':'S', 'TCT':'S',\n",
    "    'TTC':'F', 'TTT':'F', 'TTA':'L', 'TTG':'L',\n",
    "    'TAC':'Y', 'TAT':'Y', 'TAA':'_', 'TAG':'_',\n",
    "    'TGC':'C', 'TGT':'C', 'TGA':'_', 'TGG':'W',\n",
    "}\n",
    "\n",
    "def translate_dna(sequence):\n",
    "    protein = \"\"\n",
    "    for i in range(0, len(sequence), 3):\n",
    "        codon = sequence[i:i+3]\n",
    "        if len(codon) == 3:\n",
    "            protein += codon_table.get(codon, 'X')  # 'X' for unknown codons\n",
    "    return protein\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "dc34ebaa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('-',\n",
       " 'MKRTHLFIVGIYFLSSCRAEEGLNFPTYDGKDRVVSLSEKNFKQVLKKYDLLCLYYHEPVSSDKVTQKQFQLKEIVLELVAQVLEHKAIGFVMVDAKKEAKLAKKLGFDEEGSLYILKGDRTIEFDGEFAADVLVEFLLDLIEDPVEIISSKLEVQAFERIEDYIKLIGFFKSEDSEYYKAFEEAAEHFQPYIKFFATFDKGVAKKLSLKMNEVDFYEPFMDEPIAIPNKPYTEEELVEFVKEHQRPTLRRLRPEEMFETWEDDLNGIHIVAFAEKSDPDGYEFLEILKQVARDNTDNPDLSILWIDPDDFPLLVAYWEKTFKIDLFRPQIGVVNVTDADSVWMEIPDDDDLPTAEELEDWIEDVLSGXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX_',\n",
       " 'MKRTHLFIVGIYFLSSCRAEEGLNFPTYDGKDRVVSLSEKNFKQVLKKYDLLCLYYHEPVSSDKVTQKQFQLKEIVLELVAQVLEHKAIGFVMVDAKKEAKLAKKLGFDEEGSLYILKGDRTIEFDGEFAADVLVEFLLDLIEDPVEIISSKLEVQAFERIEDYIKLIGFFKSEDSEYYKAFEEAAEHFQPYIKFFATFDKGVAKKLSLKMNEVDFYEPFMDEPIAIPNKPYTEEELVEFVKEHQRPTLRRLRPEEMFETWEDDLNGIHIVAFAEKSDPDGYEFLEILKQVARDNTDNPDLSILWIDPDDFPLLVAYWEKTFKIDLFRPQIGVVNVTDADSVWMEIPDDDDLPTAEELEDWIEDVLSGKINTEDDDEDDDDDDNSDEEDNDDSDDDDDE')"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene = 'CASQ2'\n",
    "dna = ''\n",
    "\n",
    "for cds_start, cds_end, _, strand, chromosome in cds_regions_loaded[gene]:\n",
    "    if strand == '+':\n",
    "        dna += reference_genome[chromosome][cds_start-1:cds_end].seq\n",
    "    else:\n",
    "        dna += reference_genome[chromosome][cds_start-1:cds_end].reverse.complement.seq\n",
    "\n",
    "# why different? Biopython's translation matches UCSC genome browser\n",
    "strand, translate_dna(dna), translate_dna_to_protein(dna)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "c126dd66",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "flank_amount = 500\n",
    "\n",
    "sequences = []\n",
    "\n",
    "for index, variant in filtered_df.iterrows():\n",
    "    gene_name = variant['gene_name']\n",
    "    chromosome = \"chr\" + str(variant['CHROM'])\n",
    "    position = int(variant['POS'])\n",
    "    ref_allele = variant['REF']\n",
    "    alt_allele = variant['ALT']\n",
    "    label = variant['cln_significance']\n",
    "\n",
    "    cds_list = cds_regions_loaded[gene_name]  # Get the list of CDS regions for the gene\n",
    "    cds_list_sorted = sorted(cds_list, key=lambda x: x[0])  # Ensure the CDS regions are sorted\n",
    "\n",
    "\n",
    "    dna_pos = 0\n",
    "    is_within_cds = False\n",
    "    # Find the index of the CDS region that contains the variant\n",
    "    # cds_regions_loaded is sorted ascending for + strand, and descending for - strand; no need to re-sort\n",
    "    for i, cds_interval in enumerate(cds_regions_loaded[gene_name]):\n",
    "        cds_start = cds_interval[0]; cds_end = cds_interval[1]; strand = cds_interval[-2]\n",
    "        #print(gene_name, dna_pos, position, cds_interval)\n",
    "        if cds_start <= position <= cds_end:\n",
    "            if strand == '+':\n",
    "                dna_pos += position - cds_start\n",
    "            else:\n",
    "                dna_pos += cds_end - position\n",
    "            is_within_cds = True\n",
    "            break\n",
    "        dna_pos += cds_end - cds_start + 1\n",
    "    \n",
    "    prot_pos = dna_pos//3\n",
    "\n",
    "    if not is_within_cds:\n",
    "        continue\n",
    "\n",
    "    dna = ''\n",
    "    for cds_start, cds_end, _, strand, chromosome in cds_regions_loaded[gene_name]:\n",
    "        if strand == '+':\n",
    "            dna += reference_genome[chromosome][cds_start-1:cds_end].seq\n",
    "        else:\n",
    "            dna += reference_genome[chromosome][cds_start-1:cds_end].reverse.complement.seq\n",
    "\n",
    "    ref_allele_len = len(ref_allele)\n",
    "    if strand == \"+\":\n",
    "        is_ref_match = int(dna[dna_pos:(dna_pos+ref_allele_len)] == ref_allele)\n",
    "        ref_dna = dna[:dna_pos] + ref_allele + dna[dna_pos+ref_allele_len:]\n",
    "        alt_dna = dna[:dna_pos] + alt_allele + dna[dna_pos+ref_allele_len:]\n",
    "    else:\n",
    "        ref_allele = reverse_complement(ref_allele)\n",
    "        alt_allele = reverse_complement(alt_allele)\n",
    "        is_ref_match = int(dna[dna_pos:(dna_pos+ref_allele_len)] == ref_allele)\n",
    "        ref_dna = dna[:dna_pos] + ref_allele + dna[dna_pos+ref_allele_len:]\n",
    "        alt_dna = dna[:dna_pos] + alt_allele + dna[dna_pos+ref_allele_len:]\n",
    "\n",
    "    ref_prot = translate_dna_to_protein(ref_dna)\n",
    "    alt_prot = translate_dna_to_protein(alt_dna)\n",
    "    ref_prot_ = ref_prot[max(0, prot_pos-flank_amount):min(len(ref_prot), prot_pos+flank_amount)]\n",
    "    alt_prot_ = alt_prot[max(0, prot_pos-flank_amount):min(len(alt_prot), prot_pos+flank_amount)]\n",
    "    ref_dna_ = ref_dna[max(0, dna_pos-flank_amount*3):min(len(ref_dna), dna_pos+flank_amount*3)]\n",
    "    alt_dna_ = alt_dna[max(0, dna_pos-flank_amount*3):min(len(alt_dna), dna_pos+flank_amount*3)]\n",
    "\n",
    "    sequences.append({\n",
    "        'ref_prot': ref_prot_,\n",
    "        'alt_prot': alt_prot_,\n",
    "        'ref_dna': ref_dna_,\n",
    "        'alt_dna': alt_dna_,\n",
    "        'strand': strand,\n",
    "        'label': label,\n",
    "        'gene': gene_name,\n",
    "        'ref': ref_allele,\n",
    "        'alt': alt_allele,\n",
    "        'POS': position,\n",
    "        'is_ref_match': is_ref_match,\n",
    "        'molecular_consequences': variant['cln_molecular_consequence']\n",
    "    })\n",
    "\n",
    "sequences_df = pd.DataFrame(sequences)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f6c70674",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-    664\n",
       "+    221\n",
       "Name: strand, dtype: int64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequences_df.strand.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e459f177",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref_prot</th>\n",
       "      <th>alt_prot</th>\n",
       "      <th>ref_dna</th>\n",
       "      <th>alt_dna</th>\n",
       "      <th>strand</th>\n",
       "      <th>label</th>\n",
       "      <th>gene</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>POS</th>\n",
       "      <th>is_ref_match</th>\n",
       "      <th>molecular_consequences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>LFIGVIIDNFNQQKKKLGGQDIFMTEEQKKYYNAMKKLGSKKPQKP...</td>\n",
       "      <td>LFIGVIIDNFNQQKKKLGGQDIFMTEEQKKYYNAMKKLGSKKPQKP...</td>\n",
       "      <td>TCTTTATTGGTGTCATCATTGACAACTTCAACCAACAGAAGAAAAA...</td>\n",
       "      <td>TCTTTATTGGTGTCATCATTGACAACTTCAACCAACAGAAGAAAAA...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>38550481</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>VGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCINQTEGDLPL...</td>\n",
       "      <td>VGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCINQTEGDLPL...</td>\n",
       "      <td>GTGGGCGCCATCCCGTCCATCATGAACGTCCTCCTCGTCTGCCTCA...</td>\n",
       "      <td>GTGGGCGCCATCCCGTCCATCATGAACGTCCTCCTCGTCTGCCTCA...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>38550890</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>ALVGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCINQTEGDL...</td>\n",
       "      <td>ALVGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCINQTEGDL...</td>\n",
       "      <td>CCCTGGTGGGCGCCATCCCGTCCATCATGAACGTCCTCCTCGTCTG...</td>\n",
       "      <td>CCCTGGTGGGCGCCATCCCGTCCATCATGAACGTCCTCCTCGTCTG...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>38550895</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>VVNALVGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCINQTE...</td>\n",
       "      <td>VVNALVGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCINQTE...</td>\n",
       "      <td>GTGGTCAATGCCCTGGTGGGCGCCATCCCGTCCATCATGAACGTCC...</td>\n",
       "      <td>GTGGTCAATGCCCTGGTGGGCGCCATCCCGTCCATCATGAACGTCC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>38550905</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>MRVVVNALVGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCIN...</td>\n",
       "      <td>MRVVVNALVGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCIN...</td>\n",
       "      <td>ATGAGGGTGGTGGTCAATGCCCTGGTGGGCGCCATCCCGTCCATCA...</td>\n",
       "      <td>ATGAGGGTGGTGGTCAATGCCCTGGTGGGCGCCATCCCGTCCATCA...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>TG</td>\n",
       "      <td>CA</td>\n",
       "      <td>38550914</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>GMRVVVNALVGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCI...</td>\n",
       "      <td>GMRVVVNALVGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCI...</td>\n",
       "      <td>CATGAGGGTGGTGGTCAATGCCCTGGTGGGCGCCATCCCGTCCATC...</td>\n",
       "      <td>CATGAGGGTGGTGGTCAATGCCCTGGTGGGCGCCATCCCGTCCATC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>GAT</td>\n",
       "      <td>AAC</td>\n",
       "      <td>38550915</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>LRTLRALRPLRALSRFEGMRVVVNALVGAIPSIMNVLLVCLIFWLI...</td>\n",
       "      <td>LRTLRALRPLRALSRFEGMRVVVNALVGAIPSIMNVLLVCLIFWLI...</td>\n",
       "      <td>GCGGACGCTGCGTGCACTCCGTCCTCTGAGAGCTCTGTCACGATTT...</td>\n",
       "      <td>GCGGACGCTGCGTGCACTCCGTCCTCTGAGAGCTCTGTCACGATTT...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>38550966</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>SLRTLRALRPLRALSRFEGMRVVVNALVGAIPSIMNVLLVCLIFWL...</td>\n",
       "      <td>SLRTLRALRPLRALSRFEGMRVVVNALVGAIPSIMNVLLVCLIFWL...</td>\n",
       "      <td>CACTGCGGACGCTGCGTGCACTCCGTCCTCTGAGAGCTCTGTCACG...</td>\n",
       "      <td>CACTGCGGACGCTGCGTGCACTCCGTCCTCTGAGAGCTCTGTCACG...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>38550970</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>WVAYGFKKYFTNAWCWLDFLIVDVSLVSLVANTLGFAEMGPIKSLR...</td>\n",
       "      <td>WVAYGFKKYFTNAWCWLDFLIVDVSLVSLVANTLGFAEMGPIKSLR...</td>\n",
       "      <td>TGGGTGGCCTACGGCTTCAAGAAGTACTTCACCAATGCCTGGTGCT...</td>\n",
       "      <td>TGGGTGGCCTACGGCTTCAAGAAGTACTTCACCAATGCCTGGTGCT...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>38551100</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>HLLRPVMLEHPPDTTTPSEEPGGPQMLTSQAPCVDGFEEPGARQRA...</td>\n",
       "      <td>HLLRPVMLEHPPDTTTPSEEPGGPQMLTSQAPCVDGFEEPGARQRA...</td>\n",
       "      <td>CCTCCTCCGCCCTGTGATGCTAGAGCACCCGCCAGACACGACCACG...</td>\n",
       "      <td>CCTCCTCCGCCCTGTGATGCTAGAGCACCCGCCAGACACGACCACG...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>AG</td>\n",
       "      <td>CC</td>\n",
       "      <td>38579373</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>AQGQPSPGTSAPGHALHGKKNSTVDCNGVVSLLGAGDPEATSPGSH...</td>\n",
       "      <td>AQGQPSPGTSAPGHALHGKKNSTVDCNGVVSLLGAGDPEATSPGSH...</td>\n",
       "      <td>CCCAGGGACAGCCCAGTCCCGGAACCTCGGCTCCTGGCCACGCCCT...</td>\n",
       "      <td>CCCAGGGACAGCCCAGTCCCGGAACCTCGGCTCCTGGCCACGCCCT...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>GA</td>\n",
       "      <td>TC</td>\n",
       "      <td>38580944</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>SDSEDGPRAMNHLSLTRGLSRTSMKPRSSRGSIFTFRRRDLGSEAD...</td>\n",
       "      <td>SDSEDGPRAMNHLSLTRGLSRTSMKPRSSRGSIFTFRRRDLGSEAD...</td>\n",
       "      <td>CTGACTCAGAAGATGGTCCCAGAGCAATGAATCATCTCAGCCTCAC...</td>\n",
       "      <td>CTGACTCAGAAGATGGTCCCAGAGCAATGAATCATCTCAGCCTCAC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>38581169</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>CGEDRLPKSDSEDGPRAMNHLSLTRGLSRTSMKPRSSRGSIFTFRR...</td>\n",
       "      <td>CGEDRLPKSDSEDGPRAMNHLSLTRGLSRTSMKPRSSRGSIFTFRR...</td>\n",
       "      <td>TGTGGGGAGGACAGGCTCCCCAAGTCTGACTCAGAAGATGGTCCCA...</td>\n",
       "      <td>TGTGGGGAGGACAGGCTCCCCAAGTCTGACTCAGAAGATGGTCCCA...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>38581194</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>ECGEDRLPKSDSEDGPRAMNHLSLTRGLSRTSMKPRSSRGSIFTFR...</td>\n",
       "      <td>ECGEDRLPKSDSEDGPRAMNHLSLTRGLSRTSMKPRSSRGSIFTFR...</td>\n",
       "      <td>AGTGTGGGGAGGACAGGCTCCCCAAGTCTGACTCAGAAGATGGTCC...</td>\n",
       "      <td>AGTGTGGGGAGGACAGGCTCCCCAAGTCTGACTCAGAAGATGGTCC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>38581196</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>ECGEDRLPKSDSEDGPRAMNHLSLTRGLSRTSMKPRSSRGSIFTFR...</td>\n",
       "      <td>ECGEDRLPKSDSEDGPRAMNHLSLTRGLSRTSMKPRSSRGSIFTFR...</td>\n",
       "      <td>GAGTGTGGGGAGGACAGGCTCCCCAAGTCTGACTCAGAAGATGGTC...</td>\n",
       "      <td>GAGTGTGGGGAGGACAGGCTCCCCAAGTCTGACTCAGAAGATGGTC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>38581197</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>SLVKILARGFCLHAFTFLRDPWNWLDFSVIIMAYVSENIKLGNLSA...</td>\n",
       "      <td>SLVKILARGFCLHAFTFLRDPWNWLDFSVIIMAYVSENIKLGNLSA...</td>\n",
       "      <td>CTCTGGTCAAGATTCTGGCTCGAGGCTTCTGCCTGCACGCGTTCAC...</td>\n",
       "      <td>CTCTGGTCAAGATTCTGGCTCGAGGCTTCTGCCTGCACGCGTTCAC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>GC</td>\n",
       "      <td>CT</td>\n",
       "      <td>38598926</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334</th>\n",
       "      <td>LQASKKLPDLYGNPPQELIGEPLEDLDPFYSTQKTFIVLNKGKTIF...</td>\n",
       "      <td>LQASKKLPDLYGNPPQELIGEPLEDLDPFYSTQKTFIVLNKGKTIF...</td>\n",
       "      <td>TGCAGGCCTCCAAAAAGCTGCCAGATCTCTATGGCAATCCACCCCA...</td>\n",
       "      <td>TGCAGGCCTCCAAAAAGCTGCCAGATCTCTATGGCAATCCACCCCA...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>CA</td>\n",
       "      <td>TG</td>\n",
       "      <td>38603929</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>AAIEKRMAEKQARGSTTLQESREGLPEEEAPRPQLDLQASKKLPDL...</td>\n",
       "      <td>AAIEKRMAEKQARGSTTLQESREGLPEEEAPRPQLDLQASKKLPDL...</td>\n",
       "      <td>GCAGCCATCGAGAAGCGCATGGCAGAGAAGCAAGCCCGCGGCTCAA...</td>\n",
       "      <td>GCAGCCATCGAGAAGCGCATGGCAGAGAAGCAAGCCCGCGGCTCAA...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>GC</td>\n",
       "      <td>AA</td>\n",
       "      <td>38604038</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361</th>\n",
       "      <td>MANFLLPRGTSSFRRFTRESLAAIEKRMAEKQARGSTTLQESREGL...</td>\n",
       "      <td>MANFLLPRGTSSFRRFTRESLAAIEKRMAEKQARGSTTLQESREGL...</td>\n",
       "      <td>ATGGCAAACTTCCTATTACCTCGGGGCACCAGCAGCTTCCGCAGGT...</td>\n",
       "      <td>ATGGCAAACTTCCTATTACCTCGGGGCACCAGCAGCTTCCGCAGGT...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>GA</td>\n",
       "      <td>CC</td>\n",
       "      <td>38604822</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>454</th>\n",
       "      <td>MANFLLPRGTSSFRRFTRESLAAIEKRMAKKQARGSTTLQESREGL...</td>\n",
       "      <td>MANFLLPRGTSSFRRFTRESLAAIEKRMAEKQARGSTTLQESREGL...</td>\n",
       "      <td>ATGGCAAACTTCCTATTACCTCGGGGCACCAGCAGCTTCCGCAGGT...</td>\n",
       "      <td>ATGGCAAACTTCCTATTACCTCGGGGCACCAGCAGCTTCCGCAGGT...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>SCN5A</td>\n",
       "      <td>CA</td>\n",
       "      <td>TG</td>\n",
       "      <td>38633221</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>574</th>\n",
       "      <td>GLLKTARLLRLVRVARKLDRYSEYGAAVLFLLMCTFALIAHWLACI...</td>\n",
       "      <td>GLLKTARLLRLVRVARKLDRYSEYGAAVLFLLMCTFALIAHWLACI...</td>\n",
       "      <td>GgctgctgaagactgcgcggctgctgcggctggtgcGCGTGGCGCG...</td>\n",
       "      <td>GgctgctgaagactgcgcggctgctgcggctggtgcGCGTGGCGCG...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>150947415</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>575</th>\n",
       "      <td>EELIGLLKTARLLRLVRVARKLDRYSEYGAAVLFLLMCTFALIAHW...</td>\n",
       "      <td>EELIGLLKTARLLRLVRVARKLDRYSEYGAAVLFLLMCTFALIAHW...</td>\n",
       "      <td>GAGGAGCTGATCGGgctgctgaagactgcgcggctgctgcggctgg...</td>\n",
       "      <td>GAGGAGCTGATCGGgctgctgaagactgcgcggctgctgcggctgg...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>150947428</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>576</th>\n",
       "      <td>SEELIGLLKTARLLRLVRVARKLDRYSEYGAAVLFLLMCTFALIAH...</td>\n",
       "      <td>SEELIGLLKTARLLRLVRVARKLDRYSEYGAAVLFLLMCTFALIAH...</td>\n",
       "      <td>TCTGAGGAGCTGATCGGgctgctgaagactgcgcggctgctgcggc...</td>\n",
       "      <td>TCTGAGGAGCTGATCGGgctgctgaagactgcgcggctgctgcggc...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>150947431</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>609</th>\n",
       "      <td>YSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILINFR...</td>\n",
       "      <td>YSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILINFR...</td>\n",
       "      <td>ACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAGGCCCGCCTGC...</td>\n",
       "      <td>ACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAGGCCCGCCTGC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>150947791</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>610</th>\n",
       "      <td>TPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILIN...</td>\n",
       "      <td>TPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILIN...</td>\n",
       "      <td>CACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAGGCCC...</td>\n",
       "      <td>CACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAGGCCC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>150947797</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>611</th>\n",
       "      <td>FTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILI...</td>\n",
       "      <td>FTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILI...</td>\n",
       "      <td>TCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAGG...</td>\n",
       "      <td>TCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAGG...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>150947800</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>612</th>\n",
       "      <td>FTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILI...</td>\n",
       "      <td>FTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILI...</td>\n",
       "      <td>TCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAGG...</td>\n",
       "      <td>TCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAGG...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>GGG</td>\n",
       "      <td>AGC</td>\n",
       "      <td>150947800</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>613</th>\n",
       "      <td>FTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILI...</td>\n",
       "      <td>FTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILI...</td>\n",
       "      <td>TTCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAG...</td>\n",
       "      <td>TTCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAG...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>150947801</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614</th>\n",
       "      <td>VFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDIL...</td>\n",
       "      <td>VFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDIL...</td>\n",
       "      <td>TCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGA...</td>\n",
       "      <td>TCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGA...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>150947803</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615</th>\n",
       "      <td>AVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDI...</td>\n",
       "      <td>AVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDI...</td>\n",
       "      <td>CTGTCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGA...</td>\n",
       "      <td>CTGTCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGA...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>150947806</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>616</th>\n",
       "      <td>YTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIV...</td>\n",
       "      <td>YTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIV...</td>\n",
       "      <td>ACACGGCTGTCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGGA...</td>\n",
       "      <td>ACACGGCTGTCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGGA...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>150947812</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617</th>\n",
       "      <td>YTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIV...</td>\n",
       "      <td>YTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIV...</td>\n",
       "      <td>TACACGGCTGTCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGG...</td>\n",
       "      <td>TACACGGCTGTCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGG...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>150947813</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>618</th>\n",
       "      <td>YTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIV...</td>\n",
       "      <td>YTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIV...</td>\n",
       "      <td>TACACGGCTGTCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGG...</td>\n",
       "      <td>TACACGGCTGTCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGG...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>150947813</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>619</th>\n",
       "      <td>LLVIYTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDI...</td>\n",
       "      <td>LLVIYTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDI...</td>\n",
       "      <td>CTGCTGGTCATCTACACGGCTGTCTTCACACCCTACTCGGCTGCCT...</td>\n",
       "      <td>CTGCTGGTCATCTACACGGCTGTCTTCACACCCTACTCGGCTGCCT...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>150947825</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>620</th>\n",
       "      <td>WLILLLVIYTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDL...</td>\n",
       "      <td>WLILLLVIYTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDL...</td>\n",
       "      <td>GGCTCATCCTGCTGCTGGTCATCTACACGGCTGTCTTCACACCCTA...</td>\n",
       "      <td>GGCTCATCCTGCTGCTGGTCATCTACACGGCTGTCTTCACACCCTA...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>150947836</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>621</th>\n",
       "      <td>DWLILLLVIYTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVD...</td>\n",
       "      <td>DWLILLLVIYTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVD...</td>\n",
       "      <td>GACTGGCTCATCCTGCTGCTGGTCATCTACACGGCTGTCTTCACAC...</td>\n",
       "      <td>GACTGGCTCATCCTGCTGCTGGTCATCTACACGGCTGTCTTCACAC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>150947840</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631</th>\n",
       "      <td>QAPRIHRWTILHYSPFKAVWDWLILLLVIYTAVFTPYSAAFLLKET...</td>\n",
       "      <td>QAPRIHRWTILHYSPFKAVWDWLILLLVIYTAVFTPYSAAFLLKET...</td>\n",
       "      <td>GGCACCGCGCATCCACCGCTGGACCATCCTGCATTACAGCCCCTTC...</td>\n",
       "      <td>GGCACCGCGCATCCACCGCTGGACCATCCTGCATTACAGCCCCTTC...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>TC</td>\n",
       "      <td>AT</td>\n",
       "      <td>150948463</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>676</th>\n",
       "      <td>NARVENCAVIYCNDGFCELCGYSRAEVMQRPCTCDFLHGPRTQRRA...</td>\n",
       "      <td>NARVENCAVIYCNDGFCELCGYSRAEVMQRPCTCDFLHGPRTQRRA...</td>\n",
       "      <td>AACGCTCGGGTGGAGAACTGCGCCGTCATCTACTGCAACGACGGCT...</td>\n",
       "      <td>AACGCTCGGGTGGAGAACTGCGCCGTCATCTACTGCAACGACGGCT...</td>\n",
       "      <td>-</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>KCNH2</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>150951796</td>\n",
       "      <td>0</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              ref_prot  \\\n",
       "22   LFIGVIIDNFNQQKKKLGGQDIFMTEEQKKYYNAMKKLGSKKPQKP...   \n",
       "64   VGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCINQTEGDLPL...   \n",
       "65   ALVGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCINQTEGDL...   \n",
       "66   VVNALVGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCINQTE...   \n",
       "67   MRVVVNALVGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCIN...   \n",
       "68   GMRVVVNALVGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCI...   \n",
       "69   LRTLRALRPLRALSRFEGMRVVVNALVGAIPSIMNVLLVCLIFWLI...   \n",
       "70   SLRTLRALRPLRALSRFEGMRVVVNALVGAIPSIMNVLLVCLIFWL...   \n",
       "75   WVAYGFKKYFTNAWCWLDFLIVDVSLVSLVANTLGFAEMGPIKSLR...   \n",
       "195  HLLRPVMLEHPPDTTTPSEEPGGPQMLTSQAPCVDGFEEPGARQRA...   \n",
       "206  AQGQPSPGTSAPGHALHGKKNSTVDCNGVVSLLGAGDPEATSPGSH...   \n",
       "222  SDSEDGPRAMNHLSLTRGLSRTSMKPRSSRGSIFTFRRRDLGSEAD...   \n",
       "223  CGEDRLPKSDSEDGPRAMNHLSLTRGLSRTSMKPRSSRGSIFTFRR...   \n",
       "224  ECGEDRLPKSDSEDGPRAMNHLSLTRGLSRTSMKPRSSRGSIFTFR...   \n",
       "225  ECGEDRLPKSDSEDGPRAMNHLSLTRGLSRTSMKPRSSRGSIFTFR...   \n",
       "284  SLVKILARGFCLHAFTFLRDPWNWLDFSVIIMAYVSENIKLGNLSA...   \n",
       "334  LQASKKLPDLYGNPPQELIGEPLEDLDPFYSTQKTFIVLNKGKTIF...   \n",
       "353  AAIEKRMAEKQARGSTTLQESREGLPEEEAPRPQLDLQASKKLPDL...   \n",
       "361  MANFLLPRGTSSFRRFTRESLAAIEKRMAEKQARGSTTLQESREGL...   \n",
       "454  MANFLLPRGTSSFRRFTRESLAAIEKRMAKKQARGSTTLQESREGL...   \n",
       "574  GLLKTARLLRLVRVARKLDRYSEYGAAVLFLLMCTFALIAHWLACI...   \n",
       "575  EELIGLLKTARLLRLVRVARKLDRYSEYGAAVLFLLMCTFALIAHW...   \n",
       "576  SEELIGLLKTARLLRLVRVARKLDRYSEYGAAVLFLLMCTFALIAH...   \n",
       "609  YSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILINFR...   \n",
       "610  TPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILIN...   \n",
       "611  FTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILI...   \n",
       "612  FTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILI...   \n",
       "613  FTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILI...   \n",
       "614  VFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDIL...   \n",
       "615  AVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDI...   \n",
       "616  YTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIV...   \n",
       "617  YTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIV...   \n",
       "618  YTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIV...   \n",
       "619  LLVIYTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDI...   \n",
       "620  WLILLLVIYTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDL...   \n",
       "621  DWLILLLVIYTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVD...   \n",
       "631  QAPRIHRWTILHYSPFKAVWDWLILLLVIYTAVFTPYSAAFLLKET...   \n",
       "676  NARVENCAVIYCNDGFCELCGYSRAEVMQRPCTCDFLHGPRTQRRA...   \n",
       "\n",
       "                                              alt_prot  \\\n",
       "22   LFIGVIIDNFNQQKKKLGGQDIFMTEEQKKYYNAMKKLGSKKPQKP...   \n",
       "64   VGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCINQTEGDLPL...   \n",
       "65   ALVGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCINQTEGDL...   \n",
       "66   VVNALVGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCINQTE...   \n",
       "67   MRVVVNALVGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCIN...   \n",
       "68   GMRVVVNALVGAIPSIMNVLLVCLIFWLIFSIMGVNLFAGKFGRCI...   \n",
       "69   LRTLRALRPLRALSRFEGMRVVVNALVGAIPSIMNVLLVCLIFWLI...   \n",
       "70   SLRTLRALRPLRALSRFEGMRVVVNALVGAIPSIMNVLLVCLIFWL...   \n",
       "75   WVAYGFKKYFTNAWCWLDFLIVDVSLVSLVANTLGFAEMGPIKSLR...   \n",
       "195  HLLRPVMLEHPPDTTTPSEEPGGPQMLTSQAPCVDGFEEPGARQRA...   \n",
       "206  AQGQPSPGTSAPGHALHGKKNSTVDCNGVVSLLGAGDPEATSPGSH...   \n",
       "222  SDSEDGPRAMNHLSLTRGLSRTSMKPRSSRGSIFTFRRRDLGSEAD...   \n",
       "223  CGEDRLPKSDSEDGPRAMNHLSLTRGLSRTSMKPRSSRGSIFTFRR...   \n",
       "224  ECGEDRLPKSDSEDGPRAMNHLSLTRGLSRTSMKPRSSRGSIFTFR...   \n",
       "225  ECGEDRLPKSDSEDGPRAMNHLSLTRGLSRTSMKPRSSRGSIFTFR...   \n",
       "284  SLVKILARGFCLHAFTFLRDPWNWLDFSVIIMAYVSENIKLGNLSA...   \n",
       "334  LQASKKLPDLYGNPPQELIGEPLEDLDPFYSTQKTFIVLNKGKTIF...   \n",
       "353  AAIEKRMAEKQARGSTTLQESREGLPEEEAPRPQLDLQASKKLPDL...   \n",
       "361  MANFLLPRGTSSFRRFTRESLAAIEKRMAEKQARGSTTLQESREGL...   \n",
       "454  MANFLLPRGTSSFRRFTRESLAAIEKRMAEKQARGSTTLQESREGL...   \n",
       "574  GLLKTARLLRLVRVARKLDRYSEYGAAVLFLLMCTFALIAHWLACI...   \n",
       "575  EELIGLLKTARLLRLVRVARKLDRYSEYGAAVLFLLMCTFALIAHW...   \n",
       "576  SEELIGLLKTARLLRLVRVARKLDRYSEYGAAVLFLLMCTFALIAH...   \n",
       "609  YSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILINFR...   \n",
       "610  TPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILIN...   \n",
       "611  FTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILI...   \n",
       "612  FTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILI...   \n",
       "613  FTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDILI...   \n",
       "614  VFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDIL...   \n",
       "615  AVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIVDI...   \n",
       "616  YTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIV...   \n",
       "617  YTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIV...   \n",
       "618  YTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDIMFIV...   \n",
       "619  LLVIYTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDLIVDI...   \n",
       "620  WLILLLVIYTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVDL...   \n",
       "621  DWLILLLVIYTAVFTPYSAAFLLKETEEGPPATECGYACQPLAVVD...   \n",
       "631  QAPRIHRWTILHYSPFKAVWDWLILLLVIYTAVFTPYSAAFLLKET...   \n",
       "676  NARVENCAVIYCNDGFCELCGYSRAEVMQRPCTCDFLHGPRTQRRA...   \n",
       "\n",
       "                                               ref_dna  \\\n",
       "22   TCTTTATTGGTGTCATCATTGACAACTTCAACCAACAGAAGAAAAA...   \n",
       "64   GTGGGCGCCATCCCGTCCATCATGAACGTCCTCCTCGTCTGCCTCA...   \n",
       "65   CCCTGGTGGGCGCCATCCCGTCCATCATGAACGTCCTCCTCGTCTG...   \n",
       "66   GTGGTCAATGCCCTGGTGGGCGCCATCCCGTCCATCATGAACGTCC...   \n",
       "67   ATGAGGGTGGTGGTCAATGCCCTGGTGGGCGCCATCCCGTCCATCA...   \n",
       "68   CATGAGGGTGGTGGTCAATGCCCTGGTGGGCGCCATCCCGTCCATC...   \n",
       "69   GCGGACGCTGCGTGCACTCCGTCCTCTGAGAGCTCTGTCACGATTT...   \n",
       "70   CACTGCGGACGCTGCGTGCACTCCGTCCTCTGAGAGCTCTGTCACG...   \n",
       "75   TGGGTGGCCTACGGCTTCAAGAAGTACTTCACCAATGCCTGGTGCT...   \n",
       "195  CCTCCTCCGCCCTGTGATGCTAGAGCACCCGCCAGACACGACCACG...   \n",
       "206  CCCAGGGACAGCCCAGTCCCGGAACCTCGGCTCCTGGCCACGCCCT...   \n",
       "222  CTGACTCAGAAGATGGTCCCAGAGCAATGAATCATCTCAGCCTCAC...   \n",
       "223  TGTGGGGAGGACAGGCTCCCCAAGTCTGACTCAGAAGATGGTCCCA...   \n",
       "224  AGTGTGGGGAGGACAGGCTCCCCAAGTCTGACTCAGAAGATGGTCC...   \n",
       "225  GAGTGTGGGGAGGACAGGCTCCCCAAGTCTGACTCAGAAGATGGTC...   \n",
       "284  CTCTGGTCAAGATTCTGGCTCGAGGCTTCTGCCTGCACGCGTTCAC...   \n",
       "334  TGCAGGCCTCCAAAAAGCTGCCAGATCTCTATGGCAATCCACCCCA...   \n",
       "353  GCAGCCATCGAGAAGCGCATGGCAGAGAAGCAAGCCCGCGGCTCAA...   \n",
       "361  ATGGCAAACTTCCTATTACCTCGGGGCACCAGCAGCTTCCGCAGGT...   \n",
       "454  ATGGCAAACTTCCTATTACCTCGGGGCACCAGCAGCTTCCGCAGGT...   \n",
       "574  GgctgctgaagactgcgcggctgctgcggctggtgcGCGTGGCGCG...   \n",
       "575  GAGGAGCTGATCGGgctgctgaagactgcgcggctgctgcggctgg...   \n",
       "576  TCTGAGGAGCTGATCGGgctgctgaagactgcgcggctgctgcggc...   \n",
       "609  ACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAGGCCCGCCTGC...   \n",
       "610  CACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAGGCCC...   \n",
       "611  TCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAGG...   \n",
       "612  TCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAGG...   \n",
       "613  TTCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAG...   \n",
       "614  TCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGA...   \n",
       "615  CTGTCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGA...   \n",
       "616  ACACGGCTGTCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGGA...   \n",
       "617  TACACGGCTGTCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGG...   \n",
       "618  TACACGGCTGTCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGG...   \n",
       "619  CTGCTGGTCATCTACACGGCTGTCTTCACACCCTACTCGGCTGCCT...   \n",
       "620  GGCTCATCCTGCTGCTGGTCATCTACACGGCTGTCTTCACACCCTA...   \n",
       "621  GACTGGCTCATCCTGCTGCTGGTCATCTACACGGCTGTCTTCACAC...   \n",
       "631  GGCACCGCGCATCCACCGCTGGACCATCCTGCATTACAGCCCCTTC...   \n",
       "676  AACGCTCGGGTGGAGAACTGCGCCGTCATCTACTGCAACGACGGCT...   \n",
       "\n",
       "                                               alt_dna strand  \\\n",
       "22   TCTTTATTGGTGTCATCATTGACAACTTCAACCAACAGAAGAAAAA...      -   \n",
       "64   GTGGGCGCCATCCCGTCCATCATGAACGTCCTCCTCGTCTGCCTCA...      -   \n",
       "65   CCCTGGTGGGCGCCATCCCGTCCATCATGAACGTCCTCCTCGTCTG...      -   \n",
       "66   GTGGTCAATGCCCTGGTGGGCGCCATCCCGTCCATCATGAACGTCC...      -   \n",
       "67   ATGAGGGTGGTGGTCAATGCCCTGGTGGGCGCCATCCCGTCCATCA...      -   \n",
       "68   CATGAGGGTGGTGGTCAATGCCCTGGTGGGCGCCATCCCGTCCATC...      -   \n",
       "69   GCGGACGCTGCGTGCACTCCGTCCTCTGAGAGCTCTGTCACGATTT...      -   \n",
       "70   CACTGCGGACGCTGCGTGCACTCCGTCCTCTGAGAGCTCTGTCACG...      -   \n",
       "75   TGGGTGGCCTACGGCTTCAAGAAGTACTTCACCAATGCCTGGTGCT...      -   \n",
       "195  CCTCCTCCGCCCTGTGATGCTAGAGCACCCGCCAGACACGACCACG...      -   \n",
       "206  CCCAGGGACAGCCCAGTCCCGGAACCTCGGCTCCTGGCCACGCCCT...      -   \n",
       "222  CTGACTCAGAAGATGGTCCCAGAGCAATGAATCATCTCAGCCTCAC...      -   \n",
       "223  TGTGGGGAGGACAGGCTCCCCAAGTCTGACTCAGAAGATGGTCCCA...      -   \n",
       "224  AGTGTGGGGAGGACAGGCTCCCCAAGTCTGACTCAGAAGATGGTCC...      -   \n",
       "225  GAGTGTGGGGAGGACAGGCTCCCCAAGTCTGACTCAGAAGATGGTC...      -   \n",
       "284  CTCTGGTCAAGATTCTGGCTCGAGGCTTCTGCCTGCACGCGTTCAC...      -   \n",
       "334  TGCAGGCCTCCAAAAAGCTGCCAGATCTCTATGGCAATCCACCCCA...      -   \n",
       "353  GCAGCCATCGAGAAGCGCATGGCAGAGAAGCAAGCCCGCGGCTCAA...      -   \n",
       "361  ATGGCAAACTTCCTATTACCTCGGGGCACCAGCAGCTTCCGCAGGT...      -   \n",
       "454  ATGGCAAACTTCCTATTACCTCGGGGCACCAGCAGCTTCCGCAGGT...      -   \n",
       "574  GgctgctgaagactgcgcggctgctgcggctggtgcGCGTGGCGCG...      -   \n",
       "575  GAGGAGCTGATCGGgctgctgaagactgcgcggctgctgcggctgg...      -   \n",
       "576  TCTGAGGAGCTGATCGGgctgctgaagactgcgcggctgctgcggc...      -   \n",
       "609  ACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAGGCCCGCCTGC...      -   \n",
       "610  CACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAGGCCC...      -   \n",
       "611  TCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAGG...      -   \n",
       "612  TCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAGG...      -   \n",
       "613  TTCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGAAG...      -   \n",
       "614  TCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGAAGA...      -   \n",
       "615  CTGTCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGGAGACGGA...      -   \n",
       "616  ACACGGCTGTCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGGA...      -   \n",
       "617  TACACGGCTGTCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGG...      -   \n",
       "618  TACACGGCTGTCTTCACACCCTACTCGGCTGCCTTCCTGCTGAAGG...      -   \n",
       "619  CTGCTGGTCATCTACACGGCTGTCTTCACACCCTACTCGGCTGCCT...      -   \n",
       "620  GGCTCATCCTGCTGCTGGTCATCTACACGGCTGTCTTCACACCCTA...      -   \n",
       "621  GACTGGCTCATCCTGCTGCTGGTCATCTACACGGCTGTCTTCACAC...      -   \n",
       "631  GGCACCGCGCATCCACCGCTGGACCATCCTGCATTACAGCCCCTTC...      -   \n",
       "676  AACGCTCGGGTGGAGAACTGCGCCGTCATCTACTGCAACGACGGCT...      -   \n",
       "\n",
       "                      label   gene  ref  alt        POS  is_ref_match  \\\n",
       "22   Uncertain_significance  SCN5A    C    T   38550481             0   \n",
       "64   Uncertain_significance  SCN5A    G    A   38550890             0   \n",
       "65   Uncertain_significance  SCN5A    G    A   38550895             0   \n",
       "66   Uncertain_significance  SCN5A    G    A   38550905             0   \n",
       "67   Uncertain_significance  SCN5A   TG   CA   38550914             0   \n",
       "68   Uncertain_significance  SCN5A  GAT  AAC   38550915             0   \n",
       "69   Uncertain_significance  SCN5A    C    G   38550966             0   \n",
       "70   Uncertain_significance  SCN5A    T    C   38550970             0   \n",
       "75   Uncertain_significance  SCN5A    A    G   38551100             0   \n",
       "195  Uncertain_significance  SCN5A   AG   CC   38579373             0   \n",
       "206  Uncertain_significance  SCN5A   GA   TC   38580944             0   \n",
       "222  Uncertain_significance  SCN5A    C    A   38581169             0   \n",
       "223  Uncertain_significance  SCN5A    C    G   38581194             0   \n",
       "224  Uncertain_significance  SCN5A    G    A   38581196             0   \n",
       "225  Uncertain_significance  SCN5A    C    T   38581197             0   \n",
       "284  Uncertain_significance  SCN5A   GC   CT   38598926             0   \n",
       "334  Uncertain_significance  SCN5A   CA   TG   38603929             0   \n",
       "353  Uncertain_significance  SCN5A   GC   AA   38604038             0   \n",
       "361  Uncertain_significance  SCN5A   GA   CC   38604822             0   \n",
       "454  Uncertain_significance  SCN5A   CA   TG   38633221             0   \n",
       "574  Uncertain_significance  KCNH2    T    C  150947415             0   \n",
       "575  Uncertain_significance  KCNH2    C    G  150947428             0   \n",
       "576  Uncertain_significance  KCNH2    G    T  150947431             0   \n",
       "609  Uncertain_significance  KCNH2    G    T  150947791             0   \n",
       "610  Uncertain_significance  KCNH2    G    A  150947797             0   \n",
       "611  Uncertain_significance  KCNH2    G    C  150947800             0   \n",
       "612  Uncertain_significance  KCNH2  GGG  AGC  150947800             0   \n",
       "613  Uncertain_significance  KCNH2    G    T  150947801             0   \n",
       "614  Uncertain_significance  KCNH2    C    T  150947803             0   \n",
       "615  Uncertain_significance  KCNH2    G    T  150947806             0   \n",
       "616  Uncertain_significance  KCNH2    G    A  150947812             0   \n",
       "617  Uncertain_significance  KCNH2    C    T  150947813             0   \n",
       "618  Uncertain_significance  KCNH2    C    G  150947813             0   \n",
       "619  Uncertain_significance  KCNH2    G    A  150947825             0   \n",
       "620  Uncertain_significance  KCNH2    G    A  150947836             0   \n",
       "621  Uncertain_significance  KCNH2    G    C  150947840             0   \n",
       "631  Uncertain_significance  KCNH2   TC   AT  150948463             0   \n",
       "676  Uncertain_significance  KCNH2    G    C  150951796             0   \n",
       "\n",
       "                                molecular_consequences  \n",
       "22                         SO:0001583|missense_variant  \n",
       "64                         SO:0001583|missense_variant  \n",
       "65                         SO:0001583|missense_variant  \n",
       "66                         SO:0001583|missense_variant  \n",
       "67                         SO:0001583|missense_variant  \n",
       "68                         SO:0001583|missense_variant  \n",
       "69                         SO:0001583|missense_variant  \n",
       "70                         SO:0001583|missense_variant  \n",
       "75                         SO:0001583|missense_variant  \n",
       "195  SO:0001583|missense_variant,SO:0001627|intron_...  \n",
       "206                        SO:0001583|missense_variant  \n",
       "222                        SO:0001583|missense_variant  \n",
       "223                        SO:0001583|missense_variant  \n",
       "224                        SO:0001583|missense_variant  \n",
       "225                        SO:0001583|missense_variant  \n",
       "284                        SO:0001583|missense_variant  \n",
       "334                        SO:0001583|missense_variant  \n",
       "353                        SO:0001583|missense_variant  \n",
       "361                        SO:0001583|missense_variant  \n",
       "454                        SO:0001583|missense_variant  \n",
       "574                        SO:0001583|missense_variant  \n",
       "575                        SO:0001583|missense_variant  \n",
       "576                        SO:0001583|missense_variant  \n",
       "609                        SO:0001583|missense_variant  \n",
       "610                        SO:0001583|missense_variant  \n",
       "611                        SO:0001583|missense_variant  \n",
       "612                        SO:0001583|missense_variant  \n",
       "613                        SO:0001583|missense_variant  \n",
       "614                        SO:0001583|missense_variant  \n",
       "615                        SO:0001583|missense_variant  \n",
       "616                        SO:0001583|missense_variant  \n",
       "617                        SO:0001583|missense_variant  \n",
       "618                        SO:0001583|missense_variant  \n",
       "619                        SO:0001583|missense_variant  \n",
       "620                        SO:0001583|missense_variant  \n",
       "621                        SO:0001583|missense_variant  \n",
       "631                        SO:0001583|missense_variant  \n",
       "676                        SO:0001583|missense_variant  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# no ref_allele mismatch\n",
    "sequences_df[sequences_df.is_ref_match==0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "11738124",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>label</th>\n",
       "      <th>molecular_consequences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [label, molecular_consequences]\n",
       "Index: []"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# ref_prot != alt_prot unless other molecular_consequences exist\n",
    "sequences_df[sequences_df.ref_prot==sequences_df.alt_prot][['label','molecular_consequences']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "1616667c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RangeIndex(start=0, stop=98, step=1)\n",
      "RangeIndex(start=0, stop=98, step=1)\n"
     ]
    }
   ],
   "source": [
    "sequences_df['gene']\n",
    "# Assuming sequences_df is your DataFrame and 'gene' is the column you want to save\n",
    "sequences_df['gene'].to_csv('clinvar_arm.csv', index=False)\n",
    "df = pd.read_csv(\"/common/zhangz2lab/zhanh/Propath-2/PLLR_ARM.csv\")\n",
    "df['gene'] = sequences_df['gene']\n",
    "df['label'] = sequences_df['label']\n",
    "# For the DataFrame loaded from CSV\n",
    "print(df.index)\n",
    "\n",
    "# For the sequences_df DataFrame\n",
    "print(sequences_df.index)\n",
    "\n",
    "df.to_csv('/common/zhangz2lab/zhanh/Propath-2/PLLR_ARM.csv', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "8edb608a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save\n",
    "sequences_df.to_csv('/common/zhangz2lab/zhanh/esm-variants/cropped/Clinvar_ARM_uncertain_protein.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "90195f68",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ATGAAGAGAACTCACTTGTTTATTGTGGGGATTTATTTTCTGTCCTCTTGCAGGGCAGAAGAGGGGCTTAATTTCCCCACATATGATGGGAAGGACCGAGTGGTAAGTCTTTCCGAGAAGAACTTCAAGCAGGTTTTAAAGAAATATGACTTGCTTTGCCTCTACTACCATGAGCCGGTGTCTTCAGATAAGGTCACGCAAAAACAGTTCCAACTGAAAGAAATCGTGCTTGAGCTTGTGGCCCAGGTCCTTGAACATAAAGCTATAGGCTTTGTGATGGTGGATGCCAAGAAAGAAGCCAAGCTTGCCAAGAAACTGGGTTTTGATGAAGAAGGAAGCCTGTATATTCTTAAGGGTGATCGCACAATAGAGTTTGATGGCGAGTTTGCAGCTGATGTCTTGGTGGAGTTCCTCTTGGATCTAATTGAAGACCCAGTGGAGATCATCAGCAGCAAACTGGAAGTCCAAGCCTTCGAACGCATTGAAGACTACATCAAACTCATTGGCTTTTTCAAGAGTGAGGACTCAGAATACTACAAGGCTTTTGAAGAAGCAGCTGAACACTTCCAGCCTTACATCAAATTCTTTGCCACCTTTGACAAAGGGGTTGCAAAGAAATTATCTTTGAAGATGAATGAGGTTGACTTCTATGAGCCATTTATGGATGAGCCCATTGCCATCCCCAACAAACCTTACACAGAAGAGGAGCTGGTGGAGTTTGTGAAGGAACACCAAAGACCCACTCTACGTCGCCTGCGCCCAGAAGAAATGTTTGAAACATGGGAAGATGATTTGAATGGGATCCACATTGTGGCCTTTGCAGAGAAGAGTGATCCAGATGGCTACGAATTCCTGGAGATCCTGAAACAGGTTGCCCGGGACAATACTGACAACCCCGATCTGAGCATCCTGTGGATCGACCCGGACGACTTTCCTCTGCTCGTTGCCTACTGGGAGAAGACTTTCAAGATTGACCTATTCAGGCCACAGATTGGGGTGGTGAATGTCACAGATGCTGACAGTGTCTGGATGGAGATTCCAGATGATGACGATCTTCCAACTGCTGAGGAGCTGGAGGACTGGATTGAGGATGTGCTTTCTGGAaagataaacactgaagatgatgatgaagatgatgatgatgatgataattctgatgaagaggataatgatgacagtgatgacgatgatgatgaATAG\n",
      "ATGAAGAGAACTCACTTGTTTATTGTGGGGATTTATTTTCTGTCCTCTTGCAGGGCAGAAGAGGGGCTTAATTTCCCCACATATGATGGGAAGGACCGAGTGGTAAGTCTTTCCGAGAAGAACTTCAAGCAGGTTTTAAAGAAATATGACTTGCTTTGCCTCTACTACCATGAGCCGGTGTCTTCAGATAAGGTCACGCAAAAACAGTTCCAACTGAAAGAAATCGTGCTTGAGCTTGTGGCCCAGGTCCTTGAACATAAAGCTATAGGCTTTGTGATGGTGGATGCCAAGAAAGAAGCCAAGCTTGCCAAGAAACTGGGTTTTGATGAAGAAGGAAGCCTGTATATTCTTAAGGGTGATCGCACAATAGAGTTTGATGGCGAGTTTGCAGCTGATGTCTTGGTGGAGTTCCTCTTGGATCTAATTGAAGACCCAGTGGAGATCATCAGCAGCAAACTGGAAGTCCAAGCCTTCGAACGCATTGAAGACTACATCAAACTCATTGGCTTTTTCAAGAGTGAGGACTCAGAATACTACAAGGCTTTTGAAGAAGCAGCTGAACACTTCCAGCCTTACATCAAATTCTTTGCCACCTTTGACAAAGGGGTTGCAAAGAAATTATCTTTGAAGATGAATGAGGTTGACTTCTATGAGCCATTTATGGATGAGCCCATTGCCATCCCCAACAAACCTTACACAGAAGAGGAGCTGGTGGAGTTTGTGAAGGAACGCCAAAGACCCACTCTACGTCGCCTGCGCCCAGAAGAAATGTTTGAAACATGGGAAGATGATTTGAATGGGATCCACATTGTGGCCTTTGCAGAGAAGAGTGATCCAGATGGCTACGAATTCCTGGAGATCCTGAAACAGGTTGCCCGGGACAATACTGACAACCCCGATCTGAGCATCCTGTGGATCGACCCGGACGACTTTCCTCTGCTCGTTGCCTACTGGGAGAAGACTTTCAAGATTGACCTATTCAGGCCACAGATTGGGGTGGTGAATGTCACAGATGCTGACAGTGTCTGGATGGAGATTCCAGATGATGACGATCTTCCAACTGCTGAGGAGCTGGAGGACTGGATTGAGGATGTGCTTTCTGGAaagataaacactgaagatgatgatgaagatgatgatgatgatgataattctgatgaagaggataatgatgacagtgatgacgatgatgatgaATAG\n"
     ]
    }
   ],
   "source": [
    "print(sequences_df['ref_dna'].iloc[0])\n",
    "print(sequences_df['alt_dna'].iloc[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "43f0954f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequences_df['ref_prot'].iloc[0] == sequences_df['alt_prot'].iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "9e0fd638",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref_prot</th>\n",
       "      <th>alt_prot</th>\n",
       "      <th>ref_dna</th>\n",
       "      <th>alt_dna</th>\n",
       "      <th>strand</th>\n",
       "      <th>label</th>\n",
       "      <th>gene</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>POS</th>\n",
       "      <th>is_ref_match</th>\n",
       "      <th>molecular_consequences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MKRTHLFIVGIYFLSSCRAEEGLNFPTYDGKDRVVSLSEKNFKQVL...</td>\n",
       "      <td>MKRTHLFIVGIYFLSSCRAEEGLNFPTYDGKDRVVSLSEKNFKQVL...</td>\n",
       "      <td>ATGAAGAGAACTCACTTGTTTATTGTGGGGATTTATTTTCTGTCCT...</td>\n",
       "      <td>ATGAAGAGAACTCACTTGTTTATTGTGGGGATTTATTTTCTGTCCT...</td>\n",
       "      <td>-</td>\n",
       "      <td>Benign/Likely_benign</td>\n",
       "      <td>CASQ2</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>115726998</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RFSLKDTEDEVRDIIRSNIHLQGKLEDPAIRWQMALYKDLPNRTDD...</td>\n",
       "      <td>RFSLKDTEDEVRDIIRSNIHLQGKLEDPAIRWQMALYKDLPNRTDD...</td>\n",
       "      <td>CGATTTAGCCTGAAAGATACCGAGGATGAAGTACGAGATATAATCC...</td>\n",
       "      <td>CGATTTAGCCTGAAAGATACCGAGGATGAAGTACGAGATATAATCC...</td>\n",
       "      <td>+</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>RYR2</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>237783695</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKSCHDEEDDDGEEEVKSFEEKEMEKQKLLYQQARLHDRGAAEMVL...</td>\n",
       "      <td>AKSCHDEEDDDGEEEVKSFEEKEMEKQKLLYQQARLHDRGAAEMVL...</td>\n",
       "      <td>CAAAGAGTTGTCATGATGAGGAAGATGACGATGGTGAAGAGGAAGT...</td>\n",
       "      <td>CAAAGAGTTGTCATGATGAGGAAGATGACGATGGTGAAGAGGAAGT...</td>\n",
       "      <td>+</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>RYR2</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>237784299</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KKMTVKDMVTAFFSSYWSIFMTLLHFVASVFRGFFRIICSLLLGGS...</td>\n",
       "      <td>KKMTVKDMVTAFFSSYWSIFMTLLHFVASVFRGFFRIICSLLLGGS...</td>\n",
       "      <td>AAAAAGATGACCGTGAAGGACATGGTCACGGCCTTCTTTTCATCCT...</td>\n",
       "      <td>AAAAAGATGACCGTGAAGGACATGGTCACGGCCTTCTTTTCATCCT...</td>\n",
       "      <td>+</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>RYR2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>237808916</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SEDLTDLKELTEESDLLSDIFGLDLKREGGQYKLIPHNPNAGLSDL...</td>\n",
       "      <td>SEDLTDLKELTEESDLLSDIFGLDLKREGGQYKLIPHNPNAGLSDL...</td>\n",
       "      <td>CCGAGGATCTGACCGACTTAAAGGAGCTGACAGAGGAAAGTGACCT...</td>\n",
       "      <td>CCGAGGATCTGACCGACTTAAAGGAGCTGACAGAGGAAAGTGACCT...</td>\n",
       "      <td>+</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>RYR2</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>237819181</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>MGSVRTNRYSIVSSEEDGMKLATMAVANGFGNGKSKVHTRQQCRSR...</td>\n",
       "      <td>MGSVRTNRYSIVSSEEDGMKLATMAVANGFGNGKSKVHTRQQCRSR...</td>\n",
       "      <td>ATGGGCAGTGTGCGAACCAACCGCTACAGCATCGTCTCTTCAGAAG...</td>\n",
       "      <td>ATGGGCAGTGTGCGAACCAACCGCTACAGCATCGTCTCTTCAGAAG...</td>\n",
       "      <td>+</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>KCNJ2</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>70175958</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>MVVPEKEQSWIPKIFKKKTCTTFIVDSTDPGGTLCQCGRPRTAHPA...</td>\n",
       "      <td>MVVPEKEQSWIPKIFKKKTCTTFIVDSTDPGGTLCQCGRPRTAHPA...</td>\n",
       "      <td>ATGGTGGTGCCGGAGAAGGAGCAGAGCTGGATCCCCAAGATCTTCA...</td>\n",
       "      <td>ATGGTGGTGCCGGAGAAGGAGCAGAGCTGGATCCCCAAGATCTTCA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>TRPM4</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>49182690</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001623|5_prime...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...</td>\n",
       "      <td>MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...</td>\n",
       "      <td>ATGTGCATCATCTTCTTTAAGTTTGATCCTCGCCCTGTTTCCAAAA...</td>\n",
       "      <td>ATGTGCATCATCTTCTTTAAGTTTGATCCTCGCCCTGTTTCCAAAA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>TANGO2</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>20052575</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001587|nonsens...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...</td>\n",
       "      <td>MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...</td>\n",
       "      <td>ATGTGCATCATCTTCTTTAAGTTTGATCCTCGCCCTGTTTCCAAAA...</td>\n",
       "      <td>ATGTGCATCATCTTCTTTAAGTTTGATCCTCGCCCTGTTTCCAAAA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>TANGO2</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>20052581</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001587|nonsens...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...</td>\n",
       "      <td>MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...</td>\n",
       "      <td>ATGTGCATCATCTTCTTTAAGTTTGATCCTCGCCCTGTTTCCAAAA...</td>\n",
       "      <td>ATGTGCATCATCTTCTTTAAGTTTGATCCTCGCCCTGTTTCCAAAA...</td>\n",
       "      <td>+</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>TANGO2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>20061538</td>\n",
       "      <td>1</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-cod...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>98 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             ref_prot  \\\n",
       "0   MKRTHLFIVGIYFLSSCRAEEGLNFPTYDGKDRVVSLSEKNFKQVL...   \n",
       "1   RFSLKDTEDEVRDIIRSNIHLQGKLEDPAIRWQMALYKDLPNRTDD...   \n",
       "2   AKSCHDEEDDDGEEEVKSFEEKEMEKQKLLYQQARLHDRGAAEMVL...   \n",
       "3   KKMTVKDMVTAFFSSYWSIFMTLLHFVASVFRGFFRIICSLLLGGS...   \n",
       "4   SEDLTDLKELTEESDLLSDIFGLDLKREGGQYKLIPHNPNAGLSDL...   \n",
       "..                                                ...   \n",
       "93  MGSVRTNRYSIVSSEEDGMKLATMAVANGFGNGKSKVHTRQQCRSR...   \n",
       "94  MVVPEKEQSWIPKIFKKKTCTTFIVDSTDPGGTLCQCGRPRTAHPA...   \n",
       "95  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...   \n",
       "96  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...   \n",
       "97  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...   \n",
       "\n",
       "                                             alt_prot  \\\n",
       "0   MKRTHLFIVGIYFLSSCRAEEGLNFPTYDGKDRVVSLSEKNFKQVL...   \n",
       "1   RFSLKDTEDEVRDIIRSNIHLQGKLEDPAIRWQMALYKDLPNRTDD...   \n",
       "2   AKSCHDEEDDDGEEEVKSFEEKEMEKQKLLYQQARLHDRGAAEMVL...   \n",
       "3   KKMTVKDMVTAFFSSYWSIFMTLLHFVASVFRGFFRIICSLLLGGS...   \n",
       "4   SEDLTDLKELTEESDLLSDIFGLDLKREGGQYKLIPHNPNAGLSDL...   \n",
       "..                                                ...   \n",
       "93  MGSVRTNRYSIVSSEEDGMKLATMAVANGFGNGKSKVHTRQQCRSR...   \n",
       "94  MVVPEKEQSWIPKIFKKKTCTTFIVDSTDPGGTLCQCGRPRTAHPA...   \n",
       "95  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...   \n",
       "96  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...   \n",
       "97  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...   \n",
       "\n",
       "                                              ref_dna  \\\n",
       "0   ATGAAGAGAACTCACTTGTTTATTGTGGGGATTTATTTTCTGTCCT...   \n",
       "1   CGATTTAGCCTGAAAGATACCGAGGATGAAGTACGAGATATAATCC...   \n",
       "2   CAAAGAGTTGTCATGATGAGGAAGATGACGATGGTGAAGAGGAAGT...   \n",
       "3   AAAAAGATGACCGTGAAGGACATGGTCACGGCCTTCTTTTCATCCT...   \n",
       "4   CCGAGGATCTGACCGACTTAAAGGAGCTGACAGAGGAAAGTGACCT...   \n",
       "..                                                ...   \n",
       "93  ATGGGCAGTGTGCGAACCAACCGCTACAGCATCGTCTCTTCAGAAG...   \n",
       "94  ATGGTGGTGCCGGAGAAGGAGCAGAGCTGGATCCCCAAGATCTTCA...   \n",
       "95  ATGTGCATCATCTTCTTTAAGTTTGATCCTCGCCCTGTTTCCAAAA...   \n",
       "96  ATGTGCATCATCTTCTTTAAGTTTGATCCTCGCCCTGTTTCCAAAA...   \n",
       "97  ATGTGCATCATCTTCTTTAAGTTTGATCCTCGCCCTGTTTCCAAAA...   \n",
       "\n",
       "                                              alt_dna strand  \\\n",
       "0   ATGAAGAGAACTCACTTGTTTATTGTGGGGATTTATTTTCTGTCCT...      -   \n",
       "1   CGATTTAGCCTGAAAGATACCGAGGATGAAGTACGAGATATAATCC...      +   \n",
       "2   CAAAGAGTTGTCATGATGAGGAAGATGACGATGGTGAAGAGGAAGT...      +   \n",
       "3   AAAAAGATGACCGTGAAGGACATGGTCACGGCCTTCTTTTCATCCT...      +   \n",
       "4   CCGAGGATCTGACCGACTTAAAGGAGCTGACAGAGGAAAGTGACCT...      +   \n",
       "..                                                ...    ...   \n",
       "93  ATGGGCAGTGTGCGAACCAACCGCTACAGCATCGTCTCTTCAGAAG...      +   \n",
       "94  ATGGTGGTGCCGGAGAAGGAGCAGAGCTGGATCCCCAAGATCTTCA...      +   \n",
       "95  ATGTGCATCATCTTCTTTAAGTTTGATCCTCGCCCTGTTTCCAAAA...      +   \n",
       "96  ATGTGCATCATCTTCTTTAAGTTTGATCCTCGCCCTGTTTCCAAAA...      +   \n",
       "97  ATGTGCATCATCTTCTTTAAGTTTGATCCTCGCCCTGTTTCCAAAA...      +   \n",
       "\n",
       "                           label    gene ref alt        POS  is_ref_match  \\\n",
       "0           Benign/Likely_benign   CASQ2   A   G  115726998             1   \n",
       "1                     Pathogenic    RYR2   A   C  237783695             1   \n",
       "2                     Pathogenic    RYR2   C   T  237784299             1   \n",
       "3              Likely_pathogenic    RYR2   G   A  237808916             1   \n",
       "4                     Pathogenic    RYR2   C   G  237819181             1   \n",
       "..                           ...     ...  ..  ..        ...           ...   \n",
       "93  Pathogenic/Likely_pathogenic   KCNJ2   A   G   70175958             1   \n",
       "94                 Likely_benign   TRPM4   G   A   49182690             1   \n",
       "95  Pathogenic/Likely_pathogenic  TANGO2   C   T   20052575             1   \n",
       "96                    Pathogenic  TANGO2   C   T   20052581             1   \n",
       "97  Pathogenic/Likely_pathogenic  TANGO2   G   A   20061538             1   \n",
       "\n",
       "                               molecular_consequences  \n",
       "0                         SO:0001583|missense_variant  \n",
       "1                         SO:0001583|missense_variant  \n",
       "2                         SO:0001583|missense_variant  \n",
       "3                         SO:0001583|missense_variant  \n",
       "4                         SO:0001583|missense_variant  \n",
       "..                                                ...  \n",
       "93                        SO:0001583|missense_variant  \n",
       "94  SO:0001583|missense_variant,SO:0001623|5_prime...  \n",
       "95  SO:0001583|missense_variant,SO:0001587|nonsens...  \n",
       "96  SO:0001583|missense_variant,SO:0001587|nonsens...  \n",
       "97  SO:0001583|missense_variant,SO:0001619|non-cod...  \n",
       "\n",
       "[98 rows x 12 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequences_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "98c96b35",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                wt_seq  \\\n",
      "0    MARGKAKEEGSWKKFIWNSEKKEFLGRTGGSWFKILLFYVIFYGCL...   \n",
      "1    MADGGEGEDEIQFLRTDDEVVLQCTATIHKEQQKLCLAAEGFGNRL...   \n",
      "2    SAVCALGNHRVAHALCSHVDEPQLLYAIENKYMPGLLRAGYYDLLI...   \n",
      "3    QTGNNTTVNIIISTVDYLLRVQESISDFYWYYSGKDVIDEQGQRNF...   \n",
      "4    ERVYFEISESSRTQWEKPQVKESKRQFIFDVVNEGGEKEKMELFVN...   \n",
      "..                                                 ...   \n",
      "880  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...   \n",
      "881  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...   \n",
      "882  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...   \n",
      "883  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...   \n",
      "884  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...   \n",
      "\n",
      "                                               mut_seq  labels  \n",
      "0    MARGKAKEEGSWKKFIWNSEKKEFLGRTGGSWFKILLFYVIFYGCL...       1  \n",
      "1    MADGGEGEDEIQFLRTDDEVVLQCTATIHKEQQKLCLAAEGFGNRL...       1  \n",
      "2    SAVCALGNHRVAHALCSHVDEPQLLYAIENKYMPGLLRAGYYDLLI...       1  \n",
      "3    QTGNNTTVNIIISTVDYLLRVQESISDFYWYYSGKDVIDEQGQRNF...       1  \n",
      "4    ERVYFEISESSRTQWEKPQVKESKRQFIFDVVNEGGEKEKMELFVN...       1  \n",
      "..                                                 ...     ...  \n",
      "880  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...       1  \n",
      "881  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...       1  \n",
      "882  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...       1  \n",
      "883  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...       1  \n",
      "884  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...       1  \n",
      "\n",
      "[885 rows x 3 columns]\n"
     ]
    }
   ],
   "source": [
    "# Adjust the mapping to account for variations in the 'cln_significance' descriptions\n",
    "def map_significance(value):\n",
    "    value = value.lower()  # Convert to lower case to handle case variations uniformly\n",
    "    if 'benign' in value:  # This will match \"Benign\" and \"Likely_benign\"\n",
    "        return 0\n",
    "    else:  # Matches \"Pathogenic\"\n",
    "        return 1\n",
    "#sequences_df_rc['seq_a'] = sequences_df_rc['seq_a'].str.upper()\n",
    "#sequences_df_rc['seq_b'] = sequences_df_rc['seq_b'].str.upper()\n",
    "# Apply the mapping function to the 'cln_significance' column\n",
    "sequences_df['labels'] = sequences_df['label'].apply(map_significance)\n",
    "#sequences_df_rc['POS'] = sequences_df['POS']\n",
    "#sequences_df_reset = sequences_df_rc.reset_index(drop=True)\n",
    "\n",
    "# Now sequences_df_reset contains the appended 'cln_significance_mapped' column\n",
    "sequences_df_rc = sequences_df[['ref_prot','alt_prot','labels']]\n",
    "sequences_df_rc = sequences_df_rc.rename(columns={'ref_prot': 'wt_seq', 'alt_prot': 'mut_seq'})\n",
    "print(sequences_df_rc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "3d99db9b",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequences_df_rc.to_csv('/common/zhangz2lab/zhanh/esm-variants/cropped/Clinvar_uncertain_ARM_protein.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e6398ffc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0    77\n",
      "1    21\n",
      "Name: true_labels_callback, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_csv('/common/zhangz2lab/zhanh/Propath-2/PLLR_ARM.csv')\n",
    "value_counts = df['true_labels_callback'].value_counts()\n",
    "\n",
    "# Print the counts\n",
    "print(value_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "39d5a041",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>wt_seq</th>\n",
       "      <th>mut_seq</th>\n",
       "      <th>labels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MARGKAKEEGSWKKFIWNSEKKEFLGRTGGSWFKILLFYVIFYGCL...</td>\n",
       "      <td>MARGKAKEEGSWKKFIWNSEKKEFLGRTGGSWFKILLFYVIFYGCL...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MADGGEGEDEIQFLRTDDEVVLQCTATIHKEQQKLCLAAEGFGNRL...</td>\n",
       "      <td>MADGGEGEDEIQFLRTDDEVVLQCTATIHKEQQKLCLAAEGFGNRL...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SAVCALGNHRVAHALCSHVDEPQLLYAIENKYMPGLLRAGYYDLLI...</td>\n",
       "      <td>SAVCALGNHRVAHALCSHVDEPQLLYAIENKYMPGLLRAGYYDLLI...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QTGNNTTVNIIISTVDYLLRVQESISDFYWYYSGKDVIDEQGQRNF...</td>\n",
       "      <td>QTGNNTTVNIIISTVDYLLRVQESISDFYWYYSGKDVIDEQGQRNF...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERVYFEISESSRTQWEKPQVKESKRQFIFDVVNEGGEKEKMELFVN...</td>\n",
       "      <td>ERVYFEISESSRTQWEKPQVKESKRQFIFDVVNEGGEKEKMELFVN...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>880</th>\n",
       "      <td>MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...</td>\n",
       "      <td>MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>881</th>\n",
       "      <td>MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...</td>\n",
       "      <td>MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>882</th>\n",
       "      <td>MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...</td>\n",
       "      <td>MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>883</th>\n",
       "      <td>MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...</td>\n",
       "      <td>MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>884</th>\n",
       "      <td>MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...</td>\n",
       "      <td>MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>885 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                wt_seq  \\\n",
       "0    MARGKAKEEGSWKKFIWNSEKKEFLGRTGGSWFKILLFYVIFYGCL...   \n",
       "1    MADGGEGEDEIQFLRTDDEVVLQCTATIHKEQQKLCLAAEGFGNRL...   \n",
       "2    SAVCALGNHRVAHALCSHVDEPQLLYAIENKYMPGLLRAGYYDLLI...   \n",
       "3    QTGNNTTVNIIISTVDYLLRVQESISDFYWYYSGKDVIDEQGQRNF...   \n",
       "4    ERVYFEISESSRTQWEKPQVKESKRQFIFDVVNEGGEKEKMELFVN...   \n",
       "..                                                 ...   \n",
       "880  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...   \n",
       "881  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...   \n",
       "882  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...   \n",
       "883  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...   \n",
       "884  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...   \n",
       "\n",
       "                                               mut_seq  labels  \n",
       "0    MARGKAKEEGSWKKFIWNSEKKEFLGRTGGSWFKILLFYVIFYGCL...       1  \n",
       "1    MADGGEGEDEIQFLRTDDEVVLQCTATIHKEQQKLCLAAEGFGNRL...       1  \n",
       "2    SAVCALGNHRVAHALCSHVDEPQLLYAIENKYMPGLLRAGYYDLLI...       1  \n",
       "3    QTGNNTTVNIIISTVDYLLRVQESISDFYWYYSGKDVIDEQGQRNF...       1  \n",
       "4    ERVYFEISESSRTQWEKPQVKESKRQFIFDVVNEGGEKEKMELFVN...       1  \n",
       "..                                                 ...     ...  \n",
       "880  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...       1  \n",
       "881  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...       1  \n",
       "882  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...       1  \n",
       "883  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...       1  \n",
       "884  MCIIFFKFDPRPVSKNAYRLILAANRDEFYSRPSKLADFWGNNNEI...       1  \n",
       "\n",
       "[885 rows x 3 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('/common/zhangz2lab/zhanh/esm-variants/cropped/Clinvar_uncertain_ARM_protein.csv')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eff72e99",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
